<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="urwahnawaz" />

<meta name="date" content="2024-02-22" />

<title>eCLIP-targets</title>

<script src="site_libs/header-attrs-2.25/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<script src="site_libs/navigation-1.1/codefolding.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.31/datatables.js"></script>
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.13.6/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.13.6/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.2.1/css/crosstalk.min.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.2.1/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->
<style type="text/css">
.code-folding-btn { margin-bottom: 4px; }
</style>



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">UPF1-FMR1</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Exploratory QC
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="Quality-control.html">QC analysis</a>
    </li>
    <li>
      <a href="Exploratory-DEG.html">Gene expression</a>
    </li>
  </ul>
</li>
<li>
  <a href="DEG-analysis.html">DEG analysis</a>
</li>
<li>
  <a href="RNA-stability.html">RNA stability</a>
</li>
<li>
  <a href="transcript-analysis.html">Transcript analysis</a>
</li>
<li>
  <a href="eCLIP-targets.html">ENCODE eCLIP and shRNA</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/urwahnawaz/UPF1-FMR1">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">

<div class="btn-group pull-right float-right">
<button type="button" class="btn btn-default btn-xs btn-secondary btn-sm dropdown-toggle" data-toggle="dropdown" data-bs-toggle="dropdown" aria-haspopup="true" aria-expanded="false"><span>Code</span> <span class="caret"></span></button>
<ul class="dropdown-menu dropdown-menu-right" style="min-width: 50px;">
<li><a id="rmd-show-all-code" href="#">Show All Code</a></li>
<li><a id="rmd-hide-all-code" href="#">Hide All Code</a></li>
</ul>
</div>



<h1 class="title toc-ignore">eCLIP-targets</h1>
<h4 class="author">urwahnawaz</h4>
<h4 class="date">2024-02-22</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-03-28
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>UPF1-FMR1/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230923code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230923)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230923code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230923)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazUPF1FMR1tree6d5116376a57e5bf02c8d069dfa1c9358c992c6ftargetblank6d51163a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/urwahnawaz/UPF1-FMR1/tree/6d5116376a57e5bf02c8d069dfa1c9358c992c6f" target="_blank">6d51163</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomurwahnawazUPF1FMR1tree6d5116376a57e5bf02c8d069dfa1c9358c992c6ftargetblank6d51163a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/urwahnawaz/UPF1-FMR1/tree/6d5116376a57e5bf02c8d069dfa1c9358c992c6f" target="_blank">6d51163</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/
    Ignored:    analysis/Exploratory-DEG.nb.html
    Ignored:    analysis/Quality-control.nb.html

Untracked files:
    Untracked:  data/lists/LCL-gene-list/
    Untracked:  output/FRAX-DEG-RBPs-res.csv
    Untracked:  output/UPF1-DEG-RBPs-res.csv
    Untracked:  output/UPF1-LCLs-DEGs.csv
    Untracked:  output/background-RBPs-res.csv

Unstaged changes:
    Modified:   analysis/_site.yml
    Modified:   output/DEG-results.Rda

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown (<code>analysis/eCLIP-targets.Rmd</code>) and
HTML (<code>docs/eCLIP-targets.html</code>) files. If you’ve configured
a remote Git repository (see <code>?wflow_git_remote</code>), click on
the hyperlinks in the table below to view the files as they were in that
past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/UPF1-FMR1/blob/6d5116376a57e5bf02c8d069dfa1c9358c992c6f/analysis/eCLIP-targets.Rmd" target="_blank">6d51163</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-03-28
</td>
<td>
wflow_publish(c("analysis/eCLIP-targets.Rmd", "analysis/index.Rmd"))
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/UPF1-FMR1/158b7c4b29943c494609c217f0bdd19e03c194fc/docs/eCLIP-targets.html" target="_blank">158b7c4</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/urwahnawaz/UPF1-FMR1/0acdc8974f5df5ed155631eee6b7d064ce0e7199/docs/eCLIP-targets.html" target="_blank">0acdc89</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-02-23
</td>
<td>
Build site.
</td>
</tr>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/urwahnawaz/UPF1-FMR1/blob/9853d523eda8e4eabad79f3d6562d3f8d558137a/analysis/eCLIP-targets.Rmd" target="_blank">9853d52</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-02-23
</td>
<td>
wflow_publish(c("analysis/eCLIP-targets.Rmd", "analysis/index.Rmd"))
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level2">
<h2>Introduction</h2>
<pre class="r"><code>source(here::here(&quot;code/libraries.R&quot;))
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(gridExtra)
library(EnsDb.Hsapiens.v86)
library(ggrepel)
library(org.Hs.eg.db)
library(msigdbr)
library(fgsea)
library(GenomicRanges)</code></pre>
<pre class="r"><code>upf1_lcls = read.csv(&quot;output/UPF1-LCLs-DEGs.csv&quot;) %&gt;% 
  dplyr::filter(adj.P.Val &lt; 0.05)
shRNA_targets = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/Data/ENCODE_eCLIP/shRNA/ENCFF734AHK.tsv&quot;, header=TRUE) %&gt;% 
  dplyr::filter(padj &lt; 0.05) %&gt;% 
  mutate_at(.vars = &quot;id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)
bedNPs = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/Data/ENCODE_eCLIP/UPF1/ENCFF713NFZ.bed&quot;)
colnames(bedNPs) = c(&quot;chrom&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;, &quot;name&quot;, 
                     &quot;score&quot;, &quot;strand&quot;, &quot;signalValue&quot;, &quot;pValue&quot;, &quot;qValue&quot;, &quot;peak&quot;)</code></pre>
<pre class="r"><code>loj &lt;- function(a, b, out, fun = &quot;-loj&quot;) {
    call &lt;- paste(&quot;intersectBed&quot;,
                &quot;-a&quot;, a,
                &quot;-b&quot;, b,
                fun, 
                &quot;&gt;&quot;, out)
    system(call, intern = FALSE, wait = TRUE)    
    print(&quot;Complete!&quot;)
  }
  
  
  # intersect with full report
  wawb &lt;- function(a, b, out) {
    call &lt;- paste(&quot;intersectBed&quot;,
                &quot;-a&quot;, a,
                &quot;-b&quot;, b,
                &quot;-wa&quot;, &quot;-wb&quot;, 
                &quot;&gt;&quot;, out)
    system(call, intern = FALSE, wait = TRUE)    
    print(&quot;Complete!&quot;)
  }</code></pre>
</div>
<div id="shrna-to-determine-rbp-responsive-genes"
class="section level2">
<h2>shRNA to determine RBP responsive genes</h2>
<pre class="r"><code>library(VennDiagram)
x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id, 
        &quot;UPF1 DEGs K562 &quot; = shRNA_targets$id)

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-4-1">
Past versions of unnamed-chunk-4-1.png
</button>
</p>
<div id="fig-unnamed-chunk-4-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/urwahnawaz/UPF1-FMR1/blob/0acdc8974f5df5ed155631eee6b7d064ce0e7199/docs/figure/eCLIP-targets.Rmd/unnamed-chunk-4-1.png" target="_blank">0acdc89</a>
</td>
<td>
urwahnawaz
</td>
<td>
2024-02-23
</td>
</tr>
</tbody>
</table>
</div>
</div>
<pre class="r"><code>upf1_lcls %&gt;% 
  dplyr::select(SYMBOL, &quot;id&quot;= ensembl_gene_id, logFC, adj.P.Val) %&gt;% 
  inner_join(shRNA_targets %&gt;% 
                dplyr::select(id, log2FoldChange), by = &quot;id&quot;) %&gt;% 
  ggscatter(., x=&quot;logFC&quot;, y = &quot;log2FoldChange&quot;, cor.coef = TRUE, add = &quot;reg.line&quot;, 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = &quot;#EF3829&quot;,
                                                 fill = &quot;lightgray&quot;)) + 
    theme_bw() + ylab(&quot;UPF1 DEGs K562 (log2FoldChange)&quot;) + xlab(&quot;UPF1 DEGS LCLs(log2FoldChange)&quot;) +
  geom_hline(yintercept = 0, size = 0.5,lty = &quot;dashed&quot;, color = &quot;grey60&quot;) +
  geom_vline(xintercept = 0, size = 0.5, lty = &quot;dashed&quot;, color = &quot;grey60&quot;)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upf1_lcls %&gt;% 
  dplyr::select(SYMBOL, &quot;id&quot;= ensembl_gene_id, logFC, adj.P.Val) %&gt;% 
  inner_join(shRNA_targets %&gt;% 
                dplyr::select(id, log2FoldChange), by = &quot;id&quot;) %&gt;%
  DT::datatable()</code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-25391ebbaa479ba352f2" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-25391ebbaa479ba352f2">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42"],["TOM1","TMSB10","LRP12","CPPED1",null,"UBE2L6","TCF7","COL1A1","CREB5","RAP1B","SRP9","CDKN1A","GON7","LAP3","PRCP","DNASE2","PEX26","KCNQ5","TAP1","PRSS16","COL6A1","SARAF","CPNE2","SEC61G","ITM2B","RNF7","FURIN","ADGRD1","RIOX1","MAN2A2","CCDC74A","ZSCAN18","AKR1A1","HS2ST1","PASK","VIM","KDELR1","MX2","PTGES2","GLUL","EVI2B","RPS15A"],["ENSG00000100284","ENSG00000034510","ENSG00000147650","ENSG00000103381","ENSG00000240024","ENSG00000156587","ENSG00000081059","ENSG00000108821","ENSG00000146592","ENSG00000127314","ENSG00000143742","ENSG00000124762","ENSG00000170270","ENSG00000002549","ENSG00000137509","ENSG00000105612","ENSG00000215193","ENSG00000185760","ENSG00000168394","ENSG00000112812","ENSG00000142156","ENSG00000133872","ENSG00000140848","ENSG00000132432","ENSG00000136156","ENSG00000114125","ENSG00000140564","ENSG00000111452","ENSG00000170468","ENSG00000196547","ENSG00000163040","ENSG00000121413","ENSG00000117448","ENSG00000153936","ENSG00000115687","ENSG00000026025","ENSG00000105438","ENSG00000183486","ENSG00000148334","ENSG00000135821","ENSG00000185862","ENSG00000134419"],[1.40825868266497,-1.26524839015019,3.51602835807115,1.08988019816083,2.03623639528398,-0.709668295083133,2.12826414628098,1.64775286317977,3.07145336365783,-0.481520823958768,-0.566793993622592,0.6373571260127699,5.73165995902621,-0.467946501894499,-0.630913184954327,-0.67161307844058,0.814080420040377,-1.36307126532611,-0.6078019345137941,1.85294169512179,1.92347267497012,-0.637526006034661,1.35399967016557,-0.5660022058445719,-1.19302169420448,-0.461741121325697,1.29772209637132,1.73357957305626,0.638029683551255,0.868459535376209,2.03074059171786,1.0153353969308,0.379378431808373,-0.525543028902001,0.947901409271675,-1.52302644138999,-0.346235788757509,-0.844814445332523,0.538166000100144,1.70044211528327,-0.737218937190896,0.415987205305026],[9.52193675581436e-05,0.000127967158275304,0.000159869341563297,0.000392074993332969,0.00117283383550652,0.00184241159780957,0.00193860420643056,0.00433624212773328,0.00715452977584993,0.0050167527060114,0.00523296016672827,0.00528974639252593,0.00523296016672827,0.00589679587610917,0.00656510835649516,0.00664334300571776,0.00740362661889203,0.0123790155982176,0.008365745765096969,0.0125397778937518,0.0174753380602429,0.0102309976141761,0.0156009013310515,0.0112051559376786,0.0122072037092364,0.0120226380175582,0.014623632705581,0.0338727103798879,0.0222304457730274,0.0202848988400145,0.0221651283960033,0.0280079520509909,0.0249782606962441,0.0272283902722602,0.0281805200811817,0.0290745666116946,0.0356128709858227,0.0373429617102648,0.0368240458021855,0.0411252371569022,0.0411554386845136,0.0475953206345699],[-0.571190213240938,-0.578679003791449,-0.607093266549826,-0.634846951471748,-0.591463336893151,-1.10954610561804,0.665060309426649,-1.41457819152888,-1.02491164806521,0.854424611156245,0.399761202735768,-0.848452493269849,0.787420546346511,0.492570165063694,-0.41233413262338,-0.6850473064018,0.406668738470246,0.871372171099412,-0.9722112194277081,-0.848002865520472,-1.19277602802043,-0.498253173820675,-0.799892037614072,-0.474984421322497,-0.558635916398917,-0.533499548669322,-0.369050764012596,-0.582946080356194,null,-0.671354615160115,-0.675237750746563,-1.12468856340418,-0.487720214797835,0.531074138874033,0.494889902289116,-0.672234642533095,-0.664197346171513,-2.15285086100191,0.479900403231024,0.54676362901126,-1.1699769868894,-0.39166405884483]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SYMBOL<\/th>\n      <th>id<\/th>\n      <th>logFC<\/th>\n      <th>adj.P.Val<\/th>\n      <th>log2FoldChange<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,5]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"SYMBOL","targets":1},{"name":"id","targets":2},{"name":"logFC","targets":3},{"name":"adj.P.Val","targets":4},{"name":"log2FoldChange","targets":5}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
</div>
<div id="eclip-data" class="section level2">
<h2>eCLIP data</h2>
<pre class="r"><code>a= &quot;/home/neuro/Documents/NMD_analysis/Analysis/Data/ENCODE_eCLIP/UPF1/ENCFF713NFZ.bed&quot;
b= &quot;/home/neuro/Documents/NMD_analysis/Analysis/Genome/gencode.v45.chr_patch_hapl_scaff.annotation.gff3&quot;


  
out = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/Data/ENCODE_eCLIP/UPF1/overlap.txt&quot;)

colnames(out)[1:10] =  c(&quot;chrom&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;, &quot;name&quot;, 
                     &quot;score&quot;, &quot;strand&quot;, &quot;signalValue&quot;, &quot;pValue&quot;, &quot;qValue&quot;, &quot;peak&quot;)

  
out %&lt;&gt;% mutate(pval = 10^(-pValue)) %&gt;% 
  dplyr::filter(pval &lt; 0.001 &amp; signalValue &gt;=3)
  

pattern &lt;- paste0(&quot;(?&lt;=(&quot;, &quot;gene_id=&quot;, &quot;)).*?(?=(&quot;, &quot;;&quot;, &quot;))&quot;)
out %&lt;&gt;% 
  mutate(gene_id = str_extract(.$V19, pattern)) %&gt;%
  as.data.frame() %&gt;% 
  distinct(gene_id, .keep_all = TRUE) %&gt;% 
  drop_na(gene_id) %&gt;%
  mutate_at(.vars = &quot;gene_id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)


x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id, 
        &quot;UPF1 eCLIP targets K562 &quot; = out$gene_id)

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>upf1_lcls %&gt;%  
  dplyr::select(SYMBOL, &quot;gene_id&quot;= ensembl_gene_id, logFC, adj.P.Val) %&gt;%  
  inner_join(out, by = &quot;gene_id&quot;) %&gt;%
  DT::datatable() </code></pre>
<div class="datatables html-widget html-fill-item" id="htmlwidget-f61b7f79a6cd7f1c2fc4" style="width:100%;height:auto;"></div>
<script type="application/json" data-for="htmlwidget-f61b7f79a6cd7f1c2fc4">{"x":{"filter":"none","vertical":false,"data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","20","21","22","23","24","25","26","27","28","29","30","31","32","33","34","35","36","37","38","39","40","41","42","43","44","45","46","47","48","49","50","51","52","53","54","55","56","57","58","59","60","61","62","63","64","65","66","67","68","69","70","71","72","73","74","75","76","77","78","79","80","81","82","83","84","85","86","87","88","89","90","91","92","93","94"],["NRG2","SLC6A4","RGS9","FYB1","PRKCH","ZYG11A","CPPED1","KLHL6","RALY","MRPS6","PBX4","NBEAL2","SCAMP5","TCF7","LDLR","ADARB1","DGKG","CYTH1","NPL","SMARCA2","KLHL3","TSPOAP1","NLRP4","FEZ1","STRADB","CHST12","CCR7","LAP3","NFYC","PRCP","DNASE2","SPATS2","RERE","MYCL","FAM171A1","EFR3B","HIP1R","ATP6V0A2","ZNF112","AP3M2","KNTC1","PXK","KCNMB2","SNX3","IQCG","TMEM266","PTDSS2","DNAH14","KIAA0232","CCL25","MNAT1","RFC1","PMF1","ITGAE","ZNF48","PTPRS","USP4","ZNF629","SLC44A5","MTPN","DOCK4","STX18","COQ2","ALS2","TTC19","SMG6","DAPK2","ZNF652","PARP12","MUC19","HTT","CERS6","AMPD3","SCP2","DLG3","IL18BP","ABHD17B","CPEB2","SAFB","SFI1","OARD1","PTGES2","CHST11","OXR1","SLC12A8","ZMAT3","USP24","CYFIP1","AARSD1","MCM3AP","SMCO4","TEX9","PTGES3","ATP5F1B"],["ENSG00000158458","ENSG00000108576","ENSG00000108370","ENSG00000082074","ENSG00000027075","ENSG00000203995","ENSG00000103381","ENSG00000172578","ENSG00000125970","ENSG00000243927","ENSG00000105717","ENSG00000160796","ENSG00000198794","ENSG00000081059","ENSG00000130164","ENSG00000197381","ENSG00000058866","ENSG00000108669","ENSG00000135838","ENSG00000080503","ENSG00000146021","ENSG00000005379","ENSG00000160505","ENSG00000149557","ENSG00000082146","ENSG00000136213","ENSG00000126353","ENSG00000002549","ENSG00000066136","ENSG00000137509","ENSG00000105612","ENSG00000123352","ENSG00000142599","ENSG00000116990","ENSG00000148468","ENSG00000084710","ENSG00000130787","ENSG00000185344","ENSG00000062370","ENSG00000070718","ENSG00000184445","ENSG00000168297","ENSG00000197584","ENSG00000112335","ENSG00000114473","ENSG00000169758","ENSG00000174915","ENSG00000185842","ENSG00000170871","ENSG00000131142","ENSG00000020426","ENSG00000035928","ENSG00000160783","ENSG00000083457","ENSG00000180035","ENSG00000105426","ENSG00000114316","ENSG00000102870","ENSG00000137968","ENSG00000105887","ENSG00000128512","ENSG00000168818","ENSG00000173085","ENSG00000003393","ENSG00000011295","ENSG00000070366","ENSG00000035664","ENSG00000198740","ENSG00000059378","ENSG00000205592","ENSG00000197386","ENSG00000172292","ENSG00000133805","ENSG00000116171","ENSG00000082458","ENSG00000137496","ENSG00000107362","ENSG00000137449","ENSG00000160633","ENSG00000198089","ENSG00000124596","ENSG00000148334","ENSG00000171310","ENSG00000164830","ENSG00000221955","ENSG00000172667","ENSG00000162402","ENSG00000273749","ENSG00000266967","ENSG00000160294","ENSG00000166002","ENSG00000151575","ENSG00000110958","ENSG00000110955"],[3.91408248294201,2.10019398082284,2.21742641058433,3.89855121685824,4.0302592274501,2.34989840708778,1.08988019816083,2.2478079918194,-0.435115513392269,-0.783562781745288,2.26509371551107,1.01399237537007,2.33524608068991,2.12826414628098,-0.784468193129073,2.83252778705961,1.94767571909145,0.714820143473852,1.63192109351053,0.956099700847026,0.944808548508655,1.06847924122386,3.21392314709434,1.78743632869981,0.977146781938892,-1.34811819243682,1.97384167210296,-0.467946501894499,-0.372606732869284,-0.630913184954327,-0.67161307844058,-0.440914974877769,-0.893667635018083,1.19338033699631,1.64352296651472,1.82171371283674,-1.65807737661957,0.6745624448088851,1.30154274181364,0.6501099328375241,-0.6638681447834131,0.727838891501037,2.24969953946819,-0.463555768169215,0.963296451309822,1.50991989391901,0.46025838494839,2.10213605389014,0.591368694575198,-4.15651467630256,-0.379883846217497,-0.763961888215999,-0.518682092791955,-0.5617901050727589,-0.659456305356665,1.65792239518706,0.283706545315317,-1.75356934890377,4.06373150215042,-0.6010937856152,1.59518041654822,-0.520335321903435,-1.33706632764879,0.597522664788006,-0.307652520546277,0.514676245072164,1.57949747333518,-1.25171471371112,-1.05150133723396,1.99593715792914,0.599342092961282,-0.849701396697611,1.09395381543314,-0.291249926360247,-1.64233392809068,1.03525622162773,-0.415117268916346,0.630986210695905,-0.299247175456618,0.526890442818875,-0.469901629395453,0.538166000100144,-0.510709486176961,-0.6765088502908631,1.63775260804154,0.917611640009029,0.769171757144381,0.567625204352602,0.36439454480133,0.266823639492558,1.18852993918097,1.32942616394401,-0.372469784923891,0.350866431849797],[6.30141536260143e-07,1.21862950220142e-05,6.51555239425445e-05,6.38510982881364e-05,0.000100066965770269,0.000392074993332969,0.000392074993332969,0.000627242993246789,0.00106824064917041,0.00113575568184213,0.00157838680114899,0.00124961451244069,0.0021361461693063,0.00193860420643056,0.00215884088328986,0.00433624212773328,0.00477731062167917,0.0030476584463023,0.00545554577729384,0.00385232456128283,0.00523296016672827,0.00559334871075412,0.00696850604433707,0.00500103721090829,0.00576050619461566,0.00690406157718964,0.00559449931429883,0.00589679587610917,0.00626220848265134,0.00656510835649516,0.00664334300571776,0.00715452977584993,0.00740362661889203,0.0118144568015012,0.0139302160181027,0.013789941346634,0.0135048962095723,0.0111381219207362,0.0176899993689431,0.0113020619151329,0.0113935669396364,0.0127728605964562,0.0216365985067871,0.0122072037092364,0.0176899993689431,0.0231403606296658,0.0148656533727145,0.0242907254876956,0.0169401584882153,0.0132157657307965,0.0154720074198275,0.015321174467244,0.0159459324765098,0.0190082879522399,0.0276909125864516,0.0276909125864516,0.0185367519264449,0.0380655320707747,0.0377093919488392,0.0213874905230548,0.0390062682816846,0.023687175315516,0.0412074972167472,0.0356827347871338,0.0272283902722602,0.0286379144371512,0.0381930565613514,0.0393322265810244,0.0288662172239135,0.0319444648559739,0.0301401022704669,0.0338727103798879,0.0316124164648817,0.0290991208039175,0.0477214429096123,0.0338727103798879,0.0350847019734994,0.0446809643013575,0.0331464402966381,0.0346894364190501,0.0355061884561084,0.0368240458021855,0.0418682724775378,0.0399570358486294,0.0383164394663219,0.0421345291878598,0.0478731612184984,0.045587856583848,0.0464943027129826,0.047246964515364,0.0495295366739052,0.0487690607733005,0.0478003767578196,0.0484087159446955],["chr5","chr17","chr17","chr5","chr14","chr1","chr16","chr3","chr20","chr21","chr19","chr3","chr15","chr5","chr19","chr21","chr3","chr17","chr1","chr9","chr5","chr17","chr19","chr11","chr2","chr7","chr17","chr4","chr1","chr11","chr19","chr12","chr1","chr1","chr10","chr2","chr12","chr12","chr19","chr8","chr12","chr3","chr3","chr6","chr3","chr15","chr11","chr1","chr4","chr19","chr14","chr4","chr1","chr17","chr16","chr19","chr3","chr16","chr1","chr7","chr7","chr4","chr4","chr2","chr17","chr17","chr15","chr17","chr7","chr12","chr4","chr2","chr11","chr1","chrX","chr11","chr9","chr4","chr19","chr22","chr6","chr9","chr12","chr8","chr3","chr3","chr1","chr15","chr17","chr21","chr11","chr15","chr12","chr12"],[140032565,30207573,65214871,39135010,61278698,52879601,12798478,183508615,34090477,34133387,19576237,46982485,74963971,134131678,11106888,45107396,186300503,78778726,182823189,2070417,137721763,58325494,55853251,125462767,201448294,2419078,40557272,17585151,40747121,82869134,12878825,49437414,8716110,39896372,15254903,25043051,122862346,123741517,44332356,42163887,122614484,58395263,178499435,108225844,197922879,76161289,489005,225122637,6812789,8056431,60749433,39310365,156221202,3721705,30380529,5245829,49338099,30780188,75210277,135937817,111976158,4459147,83277783,201726421,16024456,2266131,63954077,49337942,140055464,40422100,3174046,168745839,10483202,52941029,70504250,72005673,71867739,15009058,5641753,31491288,41066067,128125008,104682713,106456505,125170173,179069689,55181576,22955182,42963948,46286083,93517872,56390293,56680255,56641875],[140032698,30207769,65214939,39135117,61278830,52879660,12798536,183508749,34090547,34133474,19576426,46982557,74964056,134131764,11107030,45107535,186300565,78778801,182823270,2070472,137721859,58325592,55853293,125462814,201448349,2419149,40557329,17585224,40747193,82869198,12878840,49437516,8716206,39896473,15254983,25043108,122862488,123741641,44332434,42163952,122614520,58395327,178499527,108225926,197922912,76161354,489051,225122742,6812846,8056504,60749517,39310506,156221261,3721852,30380594,5245898,49338125,30780309,75210298,135937944,111976230,4459214,83277836,201726562,16024584,2266249,63954141,49338032,140055565,40422191,3174113,168745958,10483283,52941117,70504303,72005739,71867780,15009217,5641834,31491338,41066163,128125054,104682777,106456610,125170252,179069752,55181700,22955257,42964046,46286184,93517927,56390365,56680273,56641908],["UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01","UCHL5_K562_rep01"],[1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000,1000],["+","-","-","-","+","+","-","+","+","-","+","-","-","+","+","+","+","+","+","+","+","-","+","+","+","+","-","+","+","+","-","-","-","+","-","+","-","+","+","+","+","+","-","-","+","+","+","+","+","-","+","-","-","-","-","-","-","+","+","+","+","-","-","-","-","+","-","+","-","-","+","-","+","-","+","-","+","+","+","+","+","+","+","+","-","+","+","+","-","-","+","-","-","-"],[5.82837055154946,3.52424326760336,3.40675882197666,4.21411374403426,3.09863652661433,3.55806814525162,3.83560212078053,3.32102894795078,6.03723598195018,4.92871152517202,4.05799454211698,5.66342114539785,3.42056462150169,3.79907624475542,4.00552712222285,4.8921856491469,3.26856152805664,4.51367402589317,3.55180915477994,4.60113686714351,3.16575072247286,3.1575302156679,3.32102894795078,3.04972692613338,4.68359902733548,3.35017529361029,3.01617436642235,3.01617436642235,3.97310564453047,3.57478554019656,3.97310564453047,3.58406335378457,5.1575302156679,4.16575072247286,3.21411374403426,3.42056462150169,4.42056462150169,4.32102894795077,3.42056462150169,3.90599144867193,3.60113686714351,3.21411374403426,4.32102894795077,3.43967344444939,4.42056462150169,3.32102894795078,3.09863652661433,3.64295704283814,3.21411374403426,5.1575302156679,3.32102894795078,3.26856152805664,3.90599144867193,5.42056462150169,3.21411374403426,3.51367402589317,3.32102894795078,3.83560212078053,4.09863652661433,3.21411374403426,3.38814314380931,4.09863652661433,4.56142715734154,4.97310564453047,6.8082947747848,4.90599144867193,3.98940745685957,4.06826287757081,3.03723598195018,3.32102894795078,3.32102894795078,3.79907624475542,3.32102894795078,3.05799454211698,4.14838956181143,4.42056462150169,3.62915124331311,4.90599144867193,5.02937386417721,3.32102894795078,4.68359902733548,3.32102894795078,3.09863652661433,3.38814314380931,3.97310564453047,3.83560212078053,3.60113686714351,4.03723598195018,3.19549806586692,5.1575302156679,3.83560212078053,3.42056462150169,4.09863652661433,3.83560212078053],[58.8897470826466,32.2536694463414,12.2012596050331,11.631251910564,7.93617894365629,15.8765076985773,3.01427662748618,9.72918146692108,16.2708929850391,36.1762715286742,18.25942416371,24.3817460023361,4.06757113841706,19.5205398643883,45.8690875578925,29.4004225692573,12.2681856858693,5.145647494212,27.3077260577445,10.7409480132774,9.806318970458721,6.02233930339424,3.73422919165521,6.35634996178267,5.86954014356065,22.8098084194247,4.00730967665634,4.00730967665634,3.3638520495238,51.2666903626906,3.3638520495238,10.27359068374,16.5862800623035,23.1808576434497,3.12941152952458,4.06757113841706,9.301798594885881,8.587056369228501,4.06757113841706,6.11943208215175,9.11524115909139,3.12941152952458,4.42733537566013,9.059025556008709,4.78571129132545,8.1526494800652,4.30395550647775,9.454435600065811,3.12941152952458,8.43384474358851,5.17903010558518,18.2108989686297,6.11943208215175,20.28602242984,3.12941152952458,4.40382586020835,3.73422919165521,17.6642613529949,3.71612845583689,3.12941152952458,5.76403293523502,3.71612845583689,58.4776731592453,37.4155896145825,56.8337899015232,6.96355875400114,22.8805791365579,13.6028278074358,5.41277027985128,3.73422919165521,3.73422919165521,10.8224854317738,3.73422919165521,7.63959594527006,61.1522793111518,4.78571129132545,7.10975407573368,6.96355875400114,51.8747139667063,3.73422919165521,5.86954014356065,5.17903010558518,5.71803288093237,10.3310831893599,6.46803175556678,5.77243926704013,4.74268436860049,6.81810869942724,17.0235170472696,8.43384474358851,3.01427662748618,4.06757113841706,3.71612845583689,3.01427662748618],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],[-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1,-1],["chr5","chr17","chr17","chr5","chr14","chr1","chr16","chr3","chr20","chr21","chr19","chr3","chr15","chr5","chr19","chr21","chr3","chr17","chr1","chr9","chr5","chr17","chr19","chr11","chr2","chr7","chr17","chr4","chr1","chr11","chr19","chr12","chr1","chr1","chr10","chr2","chr12","chr12","chr19","chr8","chr12","chr3","chr3","chr6","chr3","chr15","chr11","chr1","chr4","chr19","chr14","chr4","chr1","chr17","chr16","chr19","chr3","chr16","chr1","chr7","chr7","chr4","chr4","chr2","chr17","chr17","chr15","chr17","chr7","chr12","chr4","chr2","chr11","chr1","chrX","chr11","chr9","chr4","chr19","chr22","chr6","chr9","chr12","chr8","chr3","chr3","chr1","chr15","chr17","chr21","chr11","chr15","chr12","chr12"],["HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA","HAVANA"],["gene","exon","transcript","exon","gene","gene","transcript","gene","gene","transcript","gene","gene","gene","gene","transcript","gene","gene","gene","gene","exon","transcript","exon","transcript","transcript","gene","gene","gene","exon","transcript","transcript","exon","gene","transcript","gene","gene","gene","exon","transcript","gene","transcript","transcript","gene","gene","gene","gene","gene","transcript","transcript","gene","exon","transcript","transcript","transcript","transcript","gene","exon","transcript","exon","gene","gene","transcript","gene","transcript","exon","exon","exon","gene","gene","gene","exon","transcript","gene","transcript","gene","exon","gene","gene","gene","exon","transcript","transcript","exon","gene","gene","gene","gene","gene","transcript","transcript","exon","gene","transcript","gene","transcript"],[139846779,30207732,65188704,39134855,61187559,52842511,12780974,183487551,33993646,34122174,19561707,46979666,74957219,134114681,11106356,45073853,186105668,78674048,182789293,2070418,137637294,58325534,55836895,125460241,201387858,2403588,40553769,17584972,40691648,82838269,12878670,49366584,8655992,39895428,15211643,25042076,122861706,123738959,44326555,42157941,122602397,58332880,178272932,108211222,197889077,76059958,487239,225117957,6781375,8056366,60734742,39306634,156213005,3714628,30378106,5245776,49297870,30778456,75202129,135926760,111728274,4415742,83263824,201726484,16024489,2265968,63907036,49289206,140023749,40422104,3172908,168456249,10481883,52927276,70502163,71998613,71862452,15002481,5641740,31488688,41064772,128124492,104455295,106270144,125082636,178960121,55066359,22943259,42960942,46286151,93478472,56365426,56663341,56639959],[140043299,30207832,65227692,39135014,61550976,52894998,12803844,183555706,34108308,34143034,19618693,47009704,75021495,134151865,11107515,45226560,186362234,78782297,182830384,2070471,137736015,58325713,55857860,125463540,201480846,2448484,40565472,17585456,40758448,82900430,12878834,49527425,8724664,39902256,15371289,25159135,122862961,123747726,44367217,42165770,122626396,58426127,178844429,108261246,197960142,76229121,489730,225144548,6884170,8056499,60753954,39312772,156233652,3723837,30400108,5246045,49340029,30784254,75611114,135977359,112004131,4542346,83279058,201726549,16024582,2266199,64072033,49362473,140062951,40422201,3174799,168775134,10485021,53051698,70505490,72007315,71911193,15070153,5641946,31529067,41072356,128125441,104762014,106752694,125212864,179072215,55215378,22981063,42964466,46286297,93543391,56430154,56688408,56643178],[".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".","."],["-","-","+","-","+","+","-","-","+","+","-","+","+","+","+","+","-","-","+","+","-","-","+","-","+","+","-","+","+","-","-","+","-","-","-","+","+","+","-","+","+","+","+","-","-","+","+","+","+","+","+","-","+","-","+","-","-","-","-","-","-","-","-","-","+","-","-","-","-","+","+","+","+","+","+","+","-","+","+","+","-","-","+","+","-","-","-","-","-","-","-","+","-","-"],[".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".",".","."],["ID=ENSG00000158458.21;gene_id=ENSG00000158458.21;gene_type=protein_coding;gene_name=NRG2;level=2;hgnc_id=HGNC:7998;havana_gene=OTTHUMG00000129241.9","ID=exon:ENST00000650711.1:13;Parent=ENST00000650711.1;gene_id=ENSG00000108576.10;transcript_id=ENST00000650711.1;gene_type=protein_coding;gene_name=SLC6A4;transcript_type=protein_coding;transcript_name=SLC6A4-206;exon_number=13;exon_id=ENSE00003894359.1;level=2;protein_id=ENSP00000498537.1;hgnc_id=HGNC:11050;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS11256.1;havana_gene=OTTHUMG00000132751.7;havana_transcript=OTTHUMT00000504255.2","ID=ENST00000577595.1;Parent=ENSG00000108370.17;gene_id=ENSG00000108370.17;transcript_id=ENST00000577595.1;gene_type=protein_coding;gene_name=RGS9;transcript_type=retained_intron;transcript_name=RGS9-205;level=2;transcript_support_level=2;hgnc_id=HGNC:10004;havana_gene=OTTHUMG00000179349.3;havana_transcript=OTTHUMT00000445888.1","ID=exon:ENST00000512982.4:8;Parent=ENST00000512982.4;gene_id=ENSG00000082074.19;transcript_id=ENST00000512982.4;gene_type=protein_coding;gene_name=FYB1;transcript_type=protein_coding;transcript_name=FYB1-208;exon_number=8;exon_id=ENSE00000971278.1;level=2;protein_id=ENSP00000425845.3;transcript_support_level=2;hgnc_id=HGNC:4036;tag=CAGE_supported_TSS,basic,Ensembl_canonical,MANE_Select,appris_principal_4,CCDS;ccdsid=CCDS54848.1;havana_gene=OTTHUMG00000162071.10;havana_transcript=OTTHUMT00000367101.3","ID=ENSG00000027075.17;gene_id=ENSG00000027075.17;gene_type=protein_coding;gene_name=PRKCH;level=1;hgnc_id=HGNC:9403;tag=overlapping_locus;havana_gene=OTTHUMG00000152341.7","ID=ENSG00000203995.11;gene_id=ENSG00000203995.11;gene_type=protein_coding;gene_name=ZYG11A;level=2;hgnc_id=HGNC:32058;havana_gene=OTTHUMG00000008923.5","ID=ENST00000539677.1;Parent=ENSG00000103381.12;gene_id=ENSG00000103381.12;transcript_id=ENST00000539677.1;gene_type=protein_coding;gene_name=CPPED1;transcript_type=retained_intron;transcript_name=CPPED1-204;level=2;transcript_support_level=2;hgnc_id=HGNC:25632;havana_gene=OTTHUMG00000167711.4;havana_transcript=OTTHUMT00000395798.1","ID=ENSG00000172578.12;gene_id=ENSG00000172578.12;gene_type=protein_coding;gene_name=KLHL6;level=2;hgnc_id=HGNC:18653;havana_gene=OTTHUMG00000148673.3","ID=ENSG00000125970.12;gene_id=ENSG00000125970.12;gene_type=protein_coding;gene_name=RALY;level=2;hgnc_id=HGNC:15921;tag=ncRNA_host;havana_gene=OTTHUMG00000032283.4","ID=ENST00000483977.5;Parent=ENSG00000243927.6;gene_id=ENSG00000243927.6;transcript_id=ENST00000483977.5;gene_type=protein_coding;gene_name=MRPS6;transcript_type=retained_intron;transcript_name=MRPS6-204;level=2;transcript_support_level=1;hgnc_id=HGNC:14051;havana_gene=OTTHUMG00000065820.3;havana_transcript=OTTHUMT00000141034.2","ID=ENSG00000105717.14;gene_id=ENSG00000105717.14;gene_type=protein_coding;gene_name=PBX4;level=2;hgnc_id=HGNC:13403;tag=overlapping_locus;havana_gene=OTTHUMG00000172310.8","ID=ENSG00000160796.18;gene_id=ENSG00000160796.18;gene_type=protein_coding;gene_name=NBEAL2;level=1;hgnc_id=HGNC:31928;havana_gene=OTTHUMG00000156497.11","ID=ENSG00000198794.12;gene_id=ENSG00000198794.12;gene_type=protein_coding;gene_name=SCAMP5;level=2;hgnc_id=HGNC:30386;havana_gene=OTTHUMG00000172704.11","ID=ENSG00000081059.20;gene_id=ENSG00000081059.20;gene_type=protein_coding;gene_name=TCF7;level=1;hgnc_id=HGNC:11639;havana_gene=OTTHUMG00000129124.5","ID=ENST00000558528.1;Parent=ENSG00000130164.15;gene_id=ENSG00000130164.15;transcript_id=ENST00000558528.1;gene_type=protein_coding;gene_name=LDLR;transcript_type=retained_intron;transcript_name=LDLR-209;level=2;transcript_support_level=2;hgnc_id=HGNC:6547;havana_gene=OTTHUMG00000171935.9;havana_transcript=OTTHUMT00000416143.1","ID=ENSG00000197381.18;gene_id=ENSG00000197381.18;gene_type=protein_coding;gene_name=ADARB1;level=2;hgnc_id=HGNC:226;havana_gene=OTTHUMG00000090295.8","ID=ENSG00000058866.15;gene_id=ENSG00000058866.15;gene_type=protein_coding;gene_name=DGKG;level=2;hgnc_id=HGNC:2853;tag=overlapping_locus;havana_gene=OTTHUMG00000156617.6","ID=ENSG00000108669.18;gene_id=ENSG00000108669.18;gene_type=protein_coding;gene_name=CYTH1;level=1;hgnc_id=HGNC:9501;tag=ncRNA_host,overlapping_locus;havana_gene=OTTHUMG00000150253.6","ID=ENSG00000135838.14;gene_id=ENSG00000135838.14;gene_type=protein_coding;gene_name=NPL;level=2;hgnc_id=HGNC:16781;havana_gene=OTTHUMG00000035321.3","ID=exon:ENST00000357248.8:10;Parent=ENST00000357248.8;gene_id=ENSG00000080503.25;transcript_id=ENST00000357248.8;gene_type=protein_coding;gene_name=SMARCA2;transcript_type=protein_coding;transcript_name=SMARCA2-204;exon_number=10;exon_id=ENSE00001798327.1;level=2;protein_id=ENSP00000349788.2;transcript_support_level=5;hgnc_id=HGNC:11098;tag=alternative_5_UTR,not_organism_supported,RNA_Seq_supported_partial,basic,appris_alternative_2,CCDS;ccdsid=CCDS34978.1;havana_gene=OTTHUMG00000019445.14;havana_transcript=OTTHUMT00000488357.2","ID=ENST00000505853.1;Parent=ENSG00000146021.15;gene_id=ENSG00000146021.15;transcript_id=ENST00000505853.1;gene_type=protein_coding;gene_name=KLHL3;transcript_type=protein_coding;transcript_name=KLHL3-206;level=1;protein_id=ENSP00000426173.1;transcript_support_level=2;hgnc_id=HGNC:6354;tag=cds_end_NF,mRNA_end_NF,exp_conf;havana_gene=OTTHUMG00000129155.14;havana_transcript=OTTHUMT00000372234.11","ID=exon:ENST00000343736.9:4;Parent=ENST00000343736.9;gene_id=ENSG00000005379.17;transcript_id=ENST00000343736.9;gene_type=protein_coding;gene_name=TSPOAP1;transcript_type=protein_coding;transcript_name=TSPOAP1-202;exon_number=4;exon_id=ENSE00000739445.1;level=2;protein_id=ENSP00000345824.4;transcript_support_level=1;hgnc_id=HGNC:16831;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_2,CCDS;ccdsid=CCDS11605.1;havana_gene=OTTHUMG00000178923.5;havana_transcript=OTTHUMT00000443980.2","ID=ENST00000587464.1;Parent=ENSG00000160505.16;gene_id=ENSG00000160505.16;transcript_id=ENST00000587464.1;gene_type=protein_coding;gene_name=NLRP4;transcript_type=protein_coding;transcript_name=NLRP4-202;level=2;protein_id=ENSP00000468496.1;transcript_support_level=2;hgnc_id=HGNC:22943;tag=alternative_5_UTR,mRNA_end_NF,cds_end_NF;havana_gene=OTTHUMG00000181734.5;havana_transcript=OTTHUMT00000458078.2","ID=ENST00000534022.1;Parent=ENSG00000149557.14;gene_id=ENSG00000149557.14;transcript_id=ENST00000534022.1;gene_type=protein_coding;gene_name=FEZ1;transcript_type=retained_intron;transcript_name=FEZ1-215;level=2;transcript_support_level=3;hgnc_id=HGNC:3659;havana_gene=OTTHUMG00000165886.8;havana_transcript=OTTHUMT00000386883.1","ID=ENSG00000082146.13;gene_id=ENSG00000082146.13;gene_type=protein_coding;gene_name=STRADB;level=2;hgnc_id=HGNC:13205;tag=ncRNA_host;havana_gene=OTTHUMG00000132831.7","ID=ENSG00000136213.10;gene_id=ENSG00000136213.10;gene_type=protein_coding;gene_name=CHST12;level=2;hgnc_id=HGNC:17423;havana_gene=OTTHUMG00000023849.9","ID=ENSG00000126353.3;gene_id=ENSG00000126353.3;gene_type=protein_coding;gene_name=CCR7;level=2;hgnc_id=HGNC:1608;havana_gene=OTTHUMG00000133375.5","ID=exon:ENST00000509583.1:5;Parent=ENST00000509583.1;gene_id=ENSG00000002549.13;transcript_id=ENST00000509583.1;gene_type=protein_coding;gene_name=LAP3;transcript_type=protein_coding_CDS_not_defined;transcript_name=LAP3-207;exon_number=5;exon_id=ENSE00002028110.1;level=2;transcript_support_level=1;hgnc_id=HGNC:18449;tag=NMD_likely_if_extended;havana_gene=OTTHUMG00000048214.7;havana_transcript=OTTHUMT00000359988.1","ID=ENST00000467203.5;Parent=ENSG00000066136.21;gene_id=ENSG00000066136.21;transcript_id=ENST00000467203.5;gene_type=protein_coding;gene_name=NFYC;transcript_type=retained_intron;transcript_name=NFYC-216;level=2;transcript_support_level=2;hgnc_id=HGNC:7806;havana_gene=OTTHUMG00000007729.5;havana_transcript=OTTHUMT00000020808.1","ID=ENST00000531283.2;Parent=ENSG00000137509.12;gene_id=ENSG00000137509.12;transcript_id=ENST00000531283.2;gene_type=protein_coding;gene_name=PRCP;transcript_type=retained_intron;transcript_name=PRCP-210;level=2;transcript_support_level=3;hgnc_id=HGNC:9344;tag=RNA_Seq_supported_only;havana_gene=OTTHUMG00000166899.5","ID=exon:ENST00000222219.8:4;Parent=ENST00000222219.8;gene_id=ENSG00000105612.10;transcript_id=ENST00000222219.8;gene_type=protein_coding;gene_name=DNASE2;transcript_type=protein_coding;transcript_name=DNASE2-201;exon_number=4;exon_id=ENSE00003620137.1;level=2;protein_id=ENSP00000222219.2;transcript_support_level=1;hgnc_id=HGNC:2960;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS12284.1;havana_gene=OTTHUMG00000180535.3;havana_transcript=OTTHUMT00000451790.2","ID=ENSG00000123352.18;gene_id=ENSG00000123352.18;gene_type=protein_coding;gene_name=SPATS2;level=1;hgnc_id=HGNC:18650;tag=overlapping_locus;havana_gene=OTTHUMG00000169419.11","ID=ENST00000704050.1;Parent=ENSG00000142599.20;gene_id=ENSG00000142599.20;transcript_id=ENST00000704050.1;gene_type=protein_coding;gene_name=RERE;transcript_type=protein_coding;transcript_name=RERE-222;level=2;protein_id=ENSP00000515653.1;hgnc_id=HGNC:9965;tag=alternative_5_UTR,mRNA_end_NF,cds_end_NF;havana_gene=OTTHUMG00000001765.9","ID=ENSG00000116990.12;gene_id=ENSG00000116990.12;gene_type=protein_coding;gene_name=MYCL;level=2;hgnc_id=HGNC:7555;havana_gene=OTTHUMG00000009243.7","ID=ENSG00000148468.17;gene_id=ENSG00000148468.17;gene_type=protein_coding;gene_name=FAM171A1;level=2;hgnc_id=HGNC:23522;havana_gene=OTTHUMG00000017732.3","ID=ENSG00000084710.14;gene_id=ENSG00000084710.14;gene_type=protein_coding;gene_name=EFR3B;level=2;hgnc_id=HGNC:29155;tag=ncRNA_host;havana_gene=OTTHUMG00000151988.7","ID=exon:ENST00000253083.9:32;Parent=ENST00000253083.9;gene_id=ENSG00000130787.14;transcript_id=ENST00000253083.9;gene_type=protein_coding;gene_name=HIP1R;transcript_type=protein_coding;transcript_name=HIP1R-201;exon_number=32;exon_id=ENSE00001292069.4;level=2;protein_id=ENSP00000253083.4;transcript_support_level=1;hgnc_id=HGNC:18415;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS31922.1;havana_gene=OTTHUMG00000168765.2;havana_transcript=OTTHUMT00000400935.2","ID=ENST00000536426.1;Parent=ENSG00000185344.15;gene_id=ENSG00000185344.15;transcript_id=ENST00000536426.1;gene_type=protein_coding;gene_name=ATP6V0A2;transcript_type=protein_coding_CDS_not_defined;transcript_name=ATP6V0A2-204;level=2;transcript_support_level=5;hgnc_id=HGNC:18481;havana_gene=OTTHUMG00000168723.4;havana_transcript=OTTHUMT00000400769.1","ID=ENSG00000062370.17;gene_id=ENSG00000062370.17;gene_type=protein_coding;gene_name=ZNF112;level=2;hgnc_id=HGNC:12892;tag=overlapping_locus;havana_gene=OTTHUMG00000182357.4","ID=ENST00000517499.5;Parent=ENSG00000070718.13;gene_id=ENSG00000070718.13;transcript_id=ENST00000517499.5;gene_type=protein_coding;gene_name=AP3M2;transcript_type=protein_coding;transcript_name=AP3M2-202;level=1;protein_id=ENSP00000429037.1;transcript_support_level=3;hgnc_id=HGNC:570;tag=mRNA_start_NF,cds_start_NF,exp_conf;havana_gene=OTTHUMG00000164060.6;havana_transcript=OTTHUMT00000377095.1","ID=ENST00000436959.7;Parent=ENSG00000184445.12;gene_id=ENSG00000184445.12;transcript_id=ENST00000436959.7;gene_type=protein_coding;gene_name=KNTC1;transcript_type=protein_coding;transcript_name=KNTC1-204;level=2;protein_id=ENSP00000408760.2;transcript_support_level=2;hgnc_id=HGNC:17255;tag=basic;havana_gene=OTTHUMG00000168861.7;havana_transcript=OTTHUMT00000396114.2","ID=ENSG00000168297.16;gene_id=ENSG00000168297.16;gene_type=protein_coding;gene_name=PXK;level=1;hgnc_id=HGNC:23326;havana_gene=OTTHUMG00000159149.5","ID=ENSG00000197584.13;gene_id=ENSG00000197584.13;gene_type=protein_coding;gene_name=KCNMB2;level=1;hgnc_id=HGNC:6286;tag=overlapping_locus;havana_gene=OTTHUMG00000157264.6","ID=ENSG00000112335.15;gene_id=ENSG00000112335.15;gene_type=protein_coding;gene_name=SNX3;level=2;hgnc_id=HGNC:11174;havana_gene=OTTHUMG00000015323.5","ID=ENSG00000114473.14;gene_id=ENSG00000114473.14;gene_type=protein_coding;gene_name=IQCG;level=1;hgnc_id=HGNC:25251;havana_gene=OTTHUMG00000155408.3","ID=ENSG00000169758.13;gene_id=ENSG00000169758.13;gene_type=protein_coding;gene_name=TMEM266;level=1;hgnc_id=HGNC:26763;havana_gene=OTTHUMG00000142918.7","ID=ENST00000527325.1;Parent=ENSG00000174915.12;gene_id=ENSG00000174915.12;transcript_id=ENST00000527325.1;gene_type=protein_coding;gene_name=PTDSS2;transcript_type=retained_intron;transcript_name=PTDSS2-206;level=2;transcript_support_level=3;hgnc_id=HGNC:15463;havana_gene=OTTHUMG00000119087.6;havana_transcript=OTTHUMT00000384146.1","ID=ENST00000328556.9;Parent=ENSG00000185842.16;gene_id=ENSG00000185842.16;transcript_id=ENST00000328556.9;gene_type=protein_coding;gene_name=DNAH14;transcript_type=protein_coding;transcript_name=DNAH14-202;level=1;protein_id=ENSP00000332424.6;transcript_support_level=3;hgnc_id=HGNC:2945;tag=mRNA_start_NF,mRNA_end_NF,cds_start_NF,cds_end_NF,exp_conf;havana_gene=OTTHUMG00000037447.9;havana_transcript=OTTHUMT00000347361.3","ID=ENSG00000170871.12;gene_id=ENSG00000170871.12;gene_type=protein_coding;gene_name=KIAA0232;level=2;hgnc_id=HGNC:28992;havana_gene=OTTHUMG00000160072.3","ID=exon:ENST00000680646.1:5;Parent=ENST00000680646.1;gene_id=ENSG00000131142.14;transcript_id=ENST00000680646.1;gene_type=protein_coding;gene_name=CCL25;transcript_type=protein_coding;transcript_name=CCL25-208;exon_number=5;exon_id=ENSE00000898664.1;level=2;protein_id=ENSP00000504875.1;hgnc_id=HGNC:10624;tag=alternative_5_UTR,basic,appris_alternative_1,CCDS;ccdsid=CCDS12194.1;havana_gene=OTTHUMG00000141287.2","ID=ENST00000555545.1;Parent=ENSG00000020426.11;gene_id=ENSG00000020426.11;transcript_id=ENST00000555545.1;gene_type=protein_coding;gene_name=MNAT1;transcript_type=protein_coding_CDS_not_defined;transcript_name=MNAT1-206;level=1;transcript_support_level=3;hgnc_id=HGNC:7181;tag=exp_conf;havana_gene=OTTHUMG00000152340.5;havana_transcript=OTTHUMT00000412778.1","ID=ENST00000502706.1;Parent=ENSG00000035928.17;gene_id=ENSG00000035928.17;transcript_id=ENST00000502706.1;gene_type=protein_coding;gene_name=RFC1;transcript_type=protein_coding_CDS_not_defined;transcript_name=RFC1-204;level=2;transcript_support_level=5;hgnc_id=HGNC:9969;havana_gene=OTTHUMG00000099363.4;havana_transcript=OTTHUMT00000360995.1","ID=ENST00000466489.1;Parent=ENSG00000160783.19;gene_id=ENSG00000160783.19;transcript_id=ENST00000466489.1;gene_type=protein_coding;gene_name=PMF1;transcript_type=nonsense_mediated_decay;transcript_name=PMF1-204;level=2;protein_id=ENSP00000476770.1;transcript_support_level=1;hgnc_id=HGNC:9112;havana_gene=OTTHUMG00000177123.3;havana_transcript=OTTHUMT00000335578.2","ID=ENST00000572179.5;Parent=ENSG00000083457.12;gene_id=ENSG00000083457.12;transcript_id=ENST00000572179.5;gene_type=protein_coding;gene_name=ITGAE;transcript_type=protein_coding_CDS_not_defined;transcript_name=ITGAE-207;level=2;transcript_support_level=1;hgnc_id=HGNC:6147;havana_gene=OTTHUMG00000177634.4;havana_transcript=OTTHUMT00000438168.1","ID=ENSG00000180035.13;gene_id=ENSG00000180035.13;gene_type=protein_coding;gene_name=ZNF48;level=2;hgnc_id=HGNC:13114;havana_gene=OTTHUMG00000048195.7","ID=exon:ENST00000588552.5:7;Parent=ENST00000588552.5;gene_id=ENSG00000105426.19;transcript_id=ENST00000588552.5;gene_type=protein_coding;gene_name=PTPRS;transcript_type=protein_coding_CDS_not_defined;transcript_name=PTPRS-205;exon_number=7;exon_id=ENSE00003469102.1;level=2;transcript_support_level=1;hgnc_id=HGNC:9681;havana_gene=OTTHUMG00000180325.5;havana_transcript=OTTHUMT00000450761.2","ID=ENST00000486549.5;Parent=ENSG00000114316.13;gene_id=ENSG00000114316.13;transcript_id=ENST00000486549.5;gene_type=protein_coding;gene_name=USP4;transcript_type=protein_coding_CDS_not_defined;transcript_name=USP4-212;level=2;transcript_support_level=5;hgnc_id=HGNC:12627;havana_gene=OTTHUMG00000156825.3;havana_transcript=OTTHUMT00000346077.1","ID=exon:ENST00000262525.6:3;Parent=ENST00000262525.6;gene_id=ENSG00000102870.6;transcript_id=ENST00000262525.6;gene_type=protein_coding;gene_name=ZNF629;transcript_type=protein_coding;transcript_name=ZNF629-201;exon_number=3;exon_id=ENSE00000682597.6;level=2;protein_id=ENSP00000262525.4;transcript_support_level=1;hgnc_id=HGNC:29008;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS45463.1;havana_gene=OTTHUMG00000176905.3;havana_transcript=OTTHUMT00000434291.3","ID=ENSG00000137968.17;gene_id=ENSG00000137968.17;gene_type=protein_coding;gene_name=SLC44A5;level=2;hgnc_id=HGNC:28524;tag=ncRNA_host;havana_gene=OTTHUMG00000009721.3","ID=ENSG00000105887.11;gene_id=ENSG00000105887.11;gene_type=protein_coding;gene_name=MTPN;level=2;hgnc_id=HGNC:15667;tag=overlapping_locus;havana_gene=OTTHUMG00000155627.2","ID=ENST00000445943.5;Parent=ENSG00000128512.24;gene_id=ENSG00000128512.24;transcript_id=ENST00000445943.5;gene_type=protein_coding;gene_name=DOCK4;transcript_type=protein_coding;transcript_name=DOCK4-207;level=1;protein_id=ENSP00000397412.1;transcript_support_level=5;hgnc_id=HGNC:19192;tag=mRNA_start_NF,cds_start_NF,exp_conf;havana_gene=OTTHUMG00000155077.9;havana_transcript=OTTHUMT00000338942.2","ID=ENSG00000168818.10;gene_id=ENSG00000168818.10;gene_type=protein_coding;gene_name=STX18;level=1;hgnc_id=HGNC:15942;tag=overlapping_locus;havana_gene=OTTHUMG00000090331.5","ID=ENST00000503915.5;Parent=ENSG00000173085.15;gene_id=ENSG00000173085.15;transcript_id=ENST00000503915.5;gene_type=protein_coding;gene_name=COQ2;transcript_type=nonsense_mediated_decay;transcript_name=COQ2-204;level=2;protein_id=ENSP00000427146.1;transcript_support_level=1;hgnc_id=HGNC:25223;tag=mRNA_start_NF,cds_start_NF;havana_gene=OTTHUMG00000160940.6;havana_transcript=OTTHUMT00000363977.1","ID=exon:ENST00000264276.11:19;Parent=ENST00000264276.11;gene_id=ENSG00000003393.16;transcript_id=ENST00000264276.11;gene_type=protein_coding;gene_name=ALS2;transcript_type=protein_coding;transcript_name=ALS2-201;exon_number=19;exon_id=ENSE00003667933.1;level=2;protein_id=ENSP00000264276.6;transcript_support_level=1;hgnc_id=HGNC:443;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_4,CCDS;ccdsid=CCDS42800.1;havana_gene=OTTHUMG00000154507.8;havana_transcript=OTTHUMT00000335562.4","ID=exon:ENST00000497842.6:7;Parent=ENST00000497842.6;gene_id=ENSG00000011295.16;transcript_id=ENST00000497842.6;gene_type=protein_coding;gene_name=TTC19;transcript_type=protein_coding_CDS_not_defined;transcript_name=TTC19-208;exon_number=7;exon_id=ENSE00002325504.1;level=2;transcript_support_level=5;hgnc_id=HGNC:26006;havana_gene=OTTHUMG00000059306.11;havana_transcript=OTTHUMT00000131733.2","ID=exon:ENST00000354901.8:1;Parent=ENST00000354901.8;gene_id=ENSG00000070366.14;transcript_id=ENST00000354901.8;gene_type=protein_coding;gene_name=SMG6;transcript_type=protein_coding;transcript_name=SMG6-202;exon_number=1;exon_id=ENSE00002671288.1;level=2;protein_id=ENSP00000346977.4;transcript_support_level=1;hgnc_id=HGNC:17809;tag=basic,CCDS;ccdsid=CCDS58498.1;havana_gene=OTTHUMG00000177578.15;havana_transcript=OTTHUMT00000438057.1","ID=ENSG00000035664.11;gene_id=ENSG00000035664.11;gene_type=protein_coding;gene_name=DAPK2;level=2;hgnc_id=HGNC:2675;havana_gene=OTTHUMG00000132947.5","ID=ENSG00000198740.9;gene_id=ENSG00000198740.9;gene_type=protein_coding;gene_name=ZNF652;level=2;hgnc_id=HGNC:29147;havana_gene=OTTHUMG00000161322.2","ID=ENSG00000059378.13;gene_id=ENSG00000059378.13;gene_type=protein_coding;gene_name=PARP12;level=1;hgnc_id=HGNC:21919;havana_gene=OTTHUMG00000157315.2","ID=exon:ENST00000454784.10:17;Parent=ENST00000454784.10;gene_id=ENSG00000205592.16;transcript_id=ENST00000454784.10;gene_type=protein_coding;gene_name=MUC19;transcript_type=protein_coding_LoF;transcript_name=MUC19-205;exon_number=17;exon_id=ENSE00003707891.1;level=1;protein_id=ENSP00000508949.1;transcript_support_level=5;hgnc_id=HGNC:14362;tag=basic,Ensembl_canonical,appris_principal_1,exp_conf;havana_gene=OTTHUMG00000060732.36","ID=ENST00000650588.1;Parent=ENSG00000197386.14;gene_id=ENSG00000197386.14;transcript_id=ENST00000650588.1;gene_type=protein_coding;gene_name=HTT;transcript_type=nonsense_mediated_decay;transcript_name=HTT-217;level=2;protein_id=ENSP00000497844.1;hgnc_id=HGNC:4851;tag=RNA_Seq_supported_only,mRNA_start_NF,cds_start_NF;havana_gene=OTTHUMG00000159916.7;havana_transcript=OTTHUMT00000497983.1","ID=ENSG00000172292.15;gene_id=ENSG00000172292.15;gene_type=protein_coding;gene_name=CERS6;level=2;hgnc_id=HGNC:23826;tag=ncRNA_host;havana_gene=OTTHUMG00000132183.4","ID=ENST00000531227.1;Parent=ENSG00000133805.16;gene_id=ENSG00000133805.16;transcript_id=ENST00000531227.1;gene_type=protein_coding;gene_name=AMPD3;transcript_type=retained_intron;transcript_name=AMPD3-214;level=2;transcript_support_level=3;hgnc_id=HGNC:470;tag=mRNA_end_NF;havana_gene=OTTHUMG00000165682.5;havana_transcript=OTTHUMT00000385884.1","ID=ENSG00000116171.19;gene_id=ENSG00000116171.19;gene_type=protein_coding;gene_name=SCP2;level=1;hgnc_id=HGNC:10606;tag=ncRNA_host,overlapping_locus;havana_gene=OTTHUMG00000008936.6","ID=exon:ENST00000374360.8:19;Parent=ENST00000374360.8;gene_id=ENSG00000082458.12;transcript_id=ENST00000374360.8;gene_type=protein_coding;gene_name=DLG3;transcript_type=protein_coding;transcript_name=DLG3-203;exon_number=19;exon_id=ENSE00003733449.1;level=2;protein_id=ENSP00000363480.3;transcript_support_level=1;hgnc_id=HGNC:2902;tag=basic,Ensembl_canonical,MANE_Select,CCDS;ccdsid=CCDS14403.1;havana_gene=OTTHUMG00000021778.4;havana_transcript=OTTHUMT00000057074.4","ID=ENSG00000137496.19;gene_id=ENSG00000137496.19;gene_type=protein_coding;gene_name=IL18BP;level=2;hgnc_id=HGNC:5987;havana_gene=OTTHUMG00000133713.6","ID=ENSG00000107362.14;gene_id=ENSG00000107362.14;gene_type=protein_coding;gene_name=ABHD17B;level=2;hgnc_id=HGNC:24278;tag=ncRNA_host;havana_gene=OTTHUMG00000020001.1","ID=ENSG00000137449.17;gene_id=ENSG00000137449.17;gene_type=protein_coding;gene_name=CPEB2;level=2;hgnc_id=HGNC:21745;havana_gene=OTTHUMG00000090669.3","ID=exon:ENST00000588852.2:4;Parent=ENST00000588852.2;gene_id=ENSG00000160633.13;transcript_id=ENST00000588852.2;gene_type=protein_coding;gene_name=SAFB;transcript_type=protein_coding;transcript_name=SAFB-205;exon_number=4;exon_id=ENSE00003559106.1;level=2;protein_id=ENSP00000467423.1;transcript_support_level=1;hgnc_id=HGNC:10520;tag=basic,Ensembl_canonical,MANE_Select,appris_principal_5,CCDS;ccdsid=CCDS59339.1;havana_gene=OTTHUMG00000180502.4;havana_transcript=OTTHUMT00000451645.2","ID=ENST00000465646.5;Parent=ENSG00000198089.16;gene_id=ENSG00000198089.16;transcript_id=ENST00000465646.5;gene_type=protein_coding;gene_name=SFI1;transcript_type=protein_coding_CDS_not_defined;transcript_name=SFI1-216;level=1;transcript_support_level=4;hgnc_id=HGNC:29064;tag=exp_conf;havana_gene=OTTHUMG00000030249.15;havana_transcript=OTTHUMT00000315710.1","ID=ENST00000424266.7;Parent=ENSG00000124596.17;gene_id=ENSG00000124596.17;transcript_id=ENST00000424266.7;gene_type=protein_coding;gene_name=OARD1;transcript_type=protein_coding;transcript_name=OARD1-202;level=2;protein_id=ENSP00000416829.2;transcript_support_level=2;hgnc_id=HGNC:21257;tag=alternative_5_UTR,basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS34445.1;havana_gene=OTTHUMG00000014667.5;havana_transcript=OTTHUMT00000357765.2","ID=exon:ENST00000476655.6:2;Parent=ENST00000476655.6;gene_id=ENSG00000148334.16;transcript_id=ENST00000476655.6;gene_type=protein_coding;gene_name=PTGES2;transcript_type=retained_intron;transcript_name=PTGES2-205;exon_number=2;exon_id=ENSE00003905454.1;level=2;transcript_support_level=3;hgnc_id=HGNC:17822;tag=inferred_transcript_model,RNA_Seq_supported_only;havana_gene=OTTHUMG00000020730.4","ID=ENSG00000171310.11;gene_id=ENSG00000171310.11;gene_type=protein_coding;gene_name=CHST11;level=2;hgnc_id=HGNC:17422;tag=ncRNA_host,overlapping_locus;havana_gene=OTTHUMG00000169803.5","ID=ENSG00000164830.19;gene_id=ENSG00000164830.19;gene_type=protein_coding;gene_name=OXR1;level=1;hgnc_id=HGNC:15822;havana_gene=OTTHUMG00000167682.3","ID=ENSG00000221955.11;gene_id=ENSG00000221955.11;gene_type=protein_coding;gene_name=SLC12A8;level=1;hgnc_id=HGNC:15595;tag=ncRNA_host;havana_gene=OTTHUMG00000159483.10","ID=ENSG00000172667.11;gene_id=ENSG00000172667.11;gene_type=protein_coding;gene_name=ZMAT3;level=2;hgnc_id=HGNC:29983;havana_gene=OTTHUMG00000157290.7","ID=ENSG00000162402.15;gene_id=ENSG00000162402.15;gene_type=protein_coding;gene_name=USP24;level=2;hgnc_id=HGNC:12623;havana_gene=OTTHUMG00000008135.6","ID=ENST00000613006.4;Parent=ENSG00000273749.5;gene_id=ENSG00000273749.5;transcript_id=ENST00000613006.4;gene_type=protein_coding;gene_name=CYFIP1;transcript_type=protein_coding;transcript_name=CYFIP1-206;level=2;protein_id=ENSP00000480307.1;transcript_support_level=4;hgnc_id=HGNC:13759;tag=alternative_5_UTR,mRNA_end_NF,cds_end_NF;havana_gene=OTTHUMG00000129100.8;havana_transcript=OTTHUMT00000476779.2","ID=ENST00000492036.1;Parent=ENSG00000266967.7;gene_id=ENSG00000266967.7;transcript_id=ENST00000492036.1;gene_type=protein_coding;gene_name=AARSD1;transcript_type=retained_intron;transcript_name=AARSD1-214;level=2;transcript_support_level=4;hgnc_id=HGNC:28417;havana_gene=OTTHUMG00000153515.9;havana_transcript=OTTHUMT00000331681.1","ID=exon:ENST00000397708.1:1;Parent=ENST00000397708.1;gene_id=ENSG00000160294.11;transcript_id=ENST00000397708.1;gene_type=protein_coding;gene_name=MCM3AP;transcript_type=protein_coding;transcript_name=MCM3AP-202;exon_number=1;exon_id=ENSE00001529813.1;level=2;protein_id=ENSP00000380820.1;transcript_support_level=5;hgnc_id=HGNC:6946;tag=basic,appris_principal_1,CCDS;ccdsid=CCDS13734.1;havana_gene=OTTHUMG00000090630.5;havana_transcript=OTTHUMT00000207254.1","ID=ENSG00000166002.7;gene_id=ENSG00000166002.7;gene_type=protein_coding;gene_name=SMCO4;level=2;hgnc_id=HGNC:24810;havana_gene=OTTHUMG00000167442.4","ID=ENST00000696102.1;Parent=ENSG00000151575.15;gene_id=ENSG00000151575.15;transcript_id=ENST00000696102.1;gene_type=protein_coding;gene_name=TEX9;transcript_type=protein_coding;transcript_name=TEX9-210;level=2;protein_id=ENSP00000512397.1;hgnc_id=HGNC:29585;tag=RNA_Seq_supported_only,basic,Ensembl_canonical,MANE_Select,appris_principal_1,CCDS;ccdsid=CCDS10157.1;havana_gene=OTTHUMG00000132035.6","ID=ENSG00000110958.16;gene_id=ENSG00000110958.16;gene_type=protein_coding;gene_name=PTGES3;level=2;hgnc_id=HGNC:16049;tag=ncRNA_host;havana_gene=OTTHUMG00000170217.3","ID=ENST00000550162.1;Parent=ENSG00000110955.9;gene_id=ENSG00000110955.9;transcript_id=ENST00000550162.1;gene_type=protein_coding;gene_name=ATP5F1B;transcript_type=protein_coding_CDS_not_defined;transcript_name=ATP5F1B-206;level=2;transcript_support_level=2;hgnc_id=HGNC:830;tag=mRNA_end_NF;havana_gene=OTTHUMG00000170291.6;havana_transcript=OTTHUMT00000408388.1"],[1.288999999999989e-59,5.576100000000055e-33,6.291299999999467e-13,2.337480999999871e-12,1.158299999999996e-08,1.32890000000015e-16,0.0009676612999999987,1.865599999999999e-10,5.35928699999981e-17,6.663900000000082e-37,5.502700000000346e-19,4.151968000000319e-25,8.559114999999954e-05,3.016199999999863e-20,1.351800000000083e-46,3.977200000000103e-30,5.392799999999567e-13,7.15076499999995e-06,4.923500000000081e-28,1.815733000000022e-11,1.561999999999992e-10,9.498623999999962e-07,0.0001844042000000006,4.402000000000007e-07,1.350391999999998e-06,1.549499999999988e-23,9.833097000000096e-05,9.833097000000096e-05,0.0004326612000000014,5.411399999999494e-52,0.0004326612000000014,5.326100000000255e-11,2.592507000000132e-17,6.593899999999884e-24,0.0007423154000000029,8.559114999999954e-05,4.99115900000006e-10,2.587876999999992e-09,8.559114999999954e-05,7.595702000000086e-07,7.669355000000065e-10,0.0007423154000000029,3.738217999999982e-05,8.729199999999988e-10,1.63790500000001e-05,7.036400000000073e-09,4.966432000000026e-05,3.51207999999998e-10,0.0007423154000000029,3.682606000000014e-09,6.621706000000032e-06,6.153200000000136e-19,7.595702000000086e-07,5.175800999999958e-21,0.0007423154000000029,3.946154999999984e-05,0.0001844042000000006,2.166400000000254e-18,0.0001922523000000005,0.0007423154000000029,1.721737999999982e-06,0.0001922523000000005,3.329099999999863e-59,3.840700000000163e-38,1.466257000000089e-57,1.087529999999998e-07,1.316500000000045e-23,2.495584000000015e-14,3.865714000000046e-06,0.0001844042000000006,0.0001844042000000006,1.50492400000006e-11,0.0001844042000000006,2.293000000000004e-08,7.042399999999625e-62,1.63790500000001e-05,7.766867999999969e-08,1.087529999999998e-07,1.334400000000105e-52,0.0001844042000000006,1.350391999999998e-06,6.621706000000032e-06,1.914111000000008e-06,4.665700000000254e-11,3.403832999999968e-07,1.688732000000017e-06,1.808488000000016e-05,1.520166999999987e-07,9.472900000001004e-18,3.682606000000014e-09,0.0009676612999999987,8.559114999999954e-05,0.0001922523000000005,0.0009676612999999987]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>SYMBOL<\/th>\n      <th>gene_id<\/th>\n      <th>logFC<\/th>\n      <th>adj.P.Val<\/th>\n      <th>chrom<\/th>\n      <th>chromStart<\/th>\n      <th>chromEnd<\/th>\n      <th>name<\/th>\n      <th>score<\/th>\n      <th>strand<\/th>\n      <th>signalValue<\/th>\n      <th>pValue<\/th>\n      <th>qValue<\/th>\n      <th>peak<\/th>\n      <th>V11<\/th>\n      <th>V12<\/th>\n      <th>V13<\/th>\n      <th>V14<\/th>\n      <th>V15<\/th>\n      <th>V16<\/th>\n      <th>V17<\/th>\n      <th>V18<\/th>\n      <th>V19<\/th>\n      <th>pval<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"columnDefs":[{"className":"dt-right","targets":[3,4,6,7,9,11,12,13,14,18,19,24]},{"orderable":false,"targets":0},{"name":" ","targets":0},{"name":"SYMBOL","targets":1},{"name":"gene_id","targets":2},{"name":"logFC","targets":3},{"name":"adj.P.Val","targets":4},{"name":"chrom","targets":5},{"name":"chromStart","targets":6},{"name":"chromEnd","targets":7},{"name":"name","targets":8},{"name":"score","targets":9},{"name":"strand","targets":10},{"name":"signalValue","targets":11},{"name":"pValue","targets":12},{"name":"qValue","targets":13},{"name":"peak","targets":14},{"name":"V11","targets":15},{"name":"V12","targets":16},{"name":"V13","targets":17},{"name":"V14","targets":18},{"name":"V15","targets":19},{"name":"V16","targets":20},{"name":"V17","targets":21},{"name":"V18","targets":22},{"name":"V19","targets":23},{"name":"pval","targets":24}],"order":[],"autoWidth":false,"orderClasses":false},"selection":{"mode":"multiple","selected":null,"target":"row","selectable":null}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id,  
     &quot;UPF1 eCLIP targets K562 &quot; = out$gene_id,
         &quot;UPF1 DEGs K562 &quot; = shRNA_targets$id) 

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-9-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fmr1-eclip-targets-and-shrna" class="section level2">
<h2>FMR1 eCLIP targets and shRNA</h2>
<pre class="r"><code># fmr1_shRNA = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FMR1/ENCFF551WYQ.tsv&quot;,
#                         header=TRUE) %&gt;% 
#   dplyr::filter(padj &lt; 0.05) %&gt;% 
#   mutate_at(.vars = &quot;id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)

#fmr1_bed = &quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FMR1/ENCFF393YLO.bed&quot; 
#out.fmr1= &quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FMR1/overlap.txt&quot;</code></pre>
<pre class="r"><code>#loj(fmr1_bed, b, out.fmr1)


out.fmr1  = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FMR1/overlap.txt&quot;)

colnames(out.fmr1)[1:10] =  c(&quot;chrom&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;, &quot;name&quot;, 
                     &quot;score&quot;, &quot;strand&quot;, &quot;signalValue&quot;, &quot;pValue&quot;, &quot;qValue&quot;, &quot;peak&quot;)

out.fmr1 %&lt;&gt;% mutate(pval = 10^(-pValue)) %&gt;% 
  dplyr::filter(pval &lt; 0.001 &amp; signalValue &gt;=3)
  

pattern &lt;- paste0(&quot;(?&lt;=(&quot;, &quot;gene_id=&quot;, &quot;)).*?(?=(&quot;, &quot;;&quot;, &quot;))&quot;)
out.fmr1 %&lt;&gt;% 
  mutate(gene_id = str_extract(.$V19, pattern)) %&gt;%
  as.data.frame() %&gt;% 
  distinct(gene_id, .keep_all = TRUE) %&gt;% 
  drop_na(gene_id) %&gt;%
  mutate_at(.vars = &quot;gene_id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)</code></pre>
<pre class="r"><code>x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id, 
        &quot;FMR1 eCLIP targets K562 &quot; = out.fmr1$gene_id)

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-12-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id, 
        &quot;FMR1 eCLIP targets K562 &quot; = out.fmr1$gene_id, 
        &quot;UPF1 eCLIP targets K562&quot; = out$gene_id)

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-13-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
<div id="fxr1" class="section level2">
<h2>FXR1</h2>
<pre class="r"><code># fmr1_shRNA = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FMR1/ENCFF551WYQ.tsv&quot;,
#                         header=TRUE) %&gt;% 
#   dplyr::filter(padj &lt; 0.05) %&gt;% 
#   mutate_at(.vars = &quot;id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)

fxr1_bed = &quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FXR1/ENCFF136CWQ.bed&quot; 
out.fxr1 = &quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FXR1/overlap.txt&quot;</code></pre>
<pre class="r"><code>loj(fxr1_bed, b,out.fxr1)</code></pre>
<pre><code>[1] &quot;Complete!&quot;</code></pre>
<pre class="r"><code>out.fxr1  = read.table(&quot;/home/neuro/Documents/NMD_analysis/Analysis/databases/ENCODE-eCLIP/FXR1/overlap.txt&quot;)

colnames(out.fxr1)[1:10] =  c(&quot;chrom&quot;, &quot;chromStart&quot;, &quot;chromEnd&quot;, &quot;name&quot;, 
                     &quot;score&quot;, &quot;strand&quot;, &quot;signalValue&quot;, &quot;pValue&quot;, &quot;qValue&quot;, &quot;peak&quot;)

out.fxr1 %&lt;&gt;% mutate(pval = 10^(-pValue)) %&gt;% 
  dplyr::filter(pval &lt; 0.001 &amp; signalValue &gt;=3)
  

pattern &lt;- paste0(&quot;(?&lt;=(&quot;, &quot;gene_id=&quot;, &quot;)).*?(?=(&quot;, &quot;;&quot;, &quot;))&quot;)
out.fxr1 %&lt;&gt;% 
  mutate(gene_id = str_extract(.$V19, pattern)) %&gt;%
  as.data.frame() %&gt;% 
  distinct(gene_id, .keep_all = TRUE) %&gt;% 
  drop_na(gene_id) %&gt;%
  mutate_at(.vars = &quot;gene_id&quot;, .funs = gsub, pattern = &quot;\\.[0-9]*$&quot;, replacement = &quot;&quot;)</code></pre>
<pre class="r"><code>x= list(&quot;UPF1 DEGs LCLs&quot; = upf1_lcls$ensembl_gene_id, 
        &quot;FMR1 eCLIP targets K562 &quot; = out.fxr1$gene_id)

ggvenn(x)</code></pre>
<p><img src="figure/eCLIP-targets.Rmd/unnamed-chunk-17-1.png" width="672" style="display: block; margin: auto;" /></p>
<!-- # ```{r} -->
<!-- # library(ensembldb) -->
<!-- # library(ChIPseeker) -->
<!-- # library(EnsDb.Hsapiens.v86) # this pkg is about 75 Mb -->
<!-- # ah <- AnnotationHub() %>% -->
<!-- #     subset(species == "Homo sapiens") %>% -->
<!-- #     subset(rdataclass == "EnsDb") %>%  -->
<!-- #   subset(genome == "GRCh38") -->
<!-- #  -->
<!-- # ensDb <- ah[["AH109606"]] -->
<!-- # grTrans <- transcripts(ensDb) -->
<!-- # trLengths <- exonsBy(ensDb, "tx") %>% -->
<!-- #     width() %>% -->
<!-- #     vapply(sum, integer(1)) -->
<!-- # mcols(grTrans)$length <- trLengths[names(grTrans)] -->
<!-- #  -->
<!-- # genesGR = ensembldb::genes(ensDb) -->
<!-- # transGR = transcripts(ensDb) -->
<!-- #  -->
<!-- # ## significant enrichment -->
<!-- #  -->
<!-- # bedNPs %<>%  -->
<!-- #   mutate(pval = 10^(-pValue)) %>%  -->
<!-- #   dplyr::filter(pval < 0.001 & signalValue >=3) %>%  -->
<!-- #   mutate(chrom = gsub("chr", "", chrom)) -->
<!-- #    -->
<!-- #  -->
<!-- #  -->
<!-- # regions.gr = GRanges(seqnames =Rle(bedNPs$chrom) , -->
<!-- #                      ranges=IRanges(start=bedNPs$chromStart, end=bedNPs$chromEnd),  -->
<!-- #                      strand = bedNPs$strand,  -->
<!-- #                      names = bedNPs$name) -->
<!-- #  -->
<!-- # overlapGenes <- findOverlaps(genesGR, regions.gr)  -->
<!-- #  -->
<!-- # overlaps <- pintersect(genesGR[queryHits(overlapGenes)], regions.gr[subjectHits(overlapGenes)]) -->
<!-- # percentOverlap <- width(overlaps) / width(regions.gr[subjectHits(overlapGenes)]) -->
<!-- # hits <- overlapGenes[percentOverlap > 0.5] -->
<!-- #  -->
<!-- # regions.gr$gene_name <- NA -->
<!-- # regions.gr$gene_name[subjectHits(hits)] <- genesGR$gene_name[queryHits(hits)] -->
<!-- #  -->
<!-- #  -->
<!-- # genes_interact = countOverlaps(genesGR, regions.gr)%>%  -->
<!-- #   as.data.frame() %>%  -->
<!-- #   set_colnames("Overlap") %>%  -->
<!-- #   dplyr::filter(Overlap >0) -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # upf1_lcls %>%  -->
<!-- #   dplyr::select(SYMBOL, "id"= ensembl_gene_id, logFC, adj.P.Val) %>%  -->
<!-- #   inner_join(genes_interact%>%  -->
<!-- #                rownames_to_column("id"), by = "id") %>% -->
<!-- #   DT::datatable() -->
<!-- # ``` -->
<!-- #  -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # x= list("UPF1 DEGs LCLs" = upf1_lcls$ensembl_gene_id,  -->
<!-- #         "UPF1 eCLIP targets K562 " = rownames(genes_interact), -->
<!-- #          "UPF1 DEGs K562 " = shRNA_targets$id) -->
<!-- #  -->
<!-- # ggvenn(x) -->
<!-- #  -->
<!-- # ``` -->
<!-- #  -->
<!-- #  -->
<!-- # ```{r} -->
<!-- # shRNA_targets %>%  -->
<!-- #   with(table(sign(.$log2FoldChange))) -->
<!-- # ``` -->
<!-- #  -->
<!-- #  -->
<!-- #  -->
<!-- # ### -->
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.3 (2024-02-29)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.4 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/blas/libblas.so.3.10.0 
LAPACK: /usr/lib/x86_64-linux-gnu/lapack/liblapack.so.3.10.0

locale:
 [1] LC_CTYPE=en_AU.UTF-8       LC_NUMERIC=C              
 [3] LC_TIME=en_AU.UTF-8        LC_COLLATE=en_AU.UTF-8    
 [5] LC_MONETARY=en_AU.UTF-8    LC_MESSAGES=en_AU.UTF-8   
 [7] LC_PAPER=en_AU.UTF-8       LC_NAME=C                 
 [9] LC_ADDRESS=C               LC_TELEPHONE=C            
[11] LC_MEASUREMENT=en_AU.UTF-8 LC_IDENTIFICATION=C       

time zone: Australia/Adelaide
tzcode source: system (glibc)

attached base packages:
 [1] grid      stats4    tools     stats     graphics  grDevices utils    
 [8] datasets  methods   base     

other attached packages:
 [1] VennDiagram_1.7.3           futile.logger_1.4.3        
 [3] fgsea_1.28.0                msigdbr_7.5.1              
 [5] org.Hs.eg.db_3.18.0         ggrepel_0.9.5              
 [7] EnsDb.Hsapiens.v86_2.99.0   gridExtra_2.3              
 [9] naniar_1.0.0                glmpca_0.2.0               
[11] broom_1.0.5                 cowplot_1.1.2              
[13] glue_1.7.0                  ggfortify_0.4.16           
[15] stargazer_5.2.3             ngsReports_2.4.0           
[17] patchwork_1.2.0             AnnotationHub_3.10.0       
[19] BiocFileCache_2.10.1        dbplyr_2.4.0               
[21] openxlsx_4.2.5.2            ggiraph_0.8.8              
[23] DT_0.31                     msigdb_1.10.0              
[25] GSEABase_1.64.0             graph_1.80.0               
[27] annotate_1.80.0             XML_3.99-0.16              
[29] pheatmap_1.0.12             ggvenn_0.1.10              
[31] MetBrewer_0.2.0             ggpubr_0.6.0               
[33] venn_1.12                   viridis_0.6.4              
[35] viridisLite_0.4.2           tximeta_1.20.2             
[37] tximport_1.30.0             goseq_1.54.0               
[39] geneLenDataBase_1.38.0      BiasedUrn_2.0.11           
[41] org.Mm.eg.db_3.18.0         EnsDb.Mmusculus.v79_2.99.0 
[43] ensembldb_2.26.0            AnnotationFilter_1.26.0    
[45] GenomicFeatures_1.54.1      AnnotationDbi_1.64.1       
[47] biomaRt_2.58.0              edgeR_4.0.11               
[49] limma_3.58.1                DESeq2_1.42.0              
[51] SummarizedExperiment_1.32.0 Biobase_2.62.0             
[53] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[55] GenomicRanges_1.54.1        GenomeInfoDb_1.38.5        
[57] IRanges_2.36.0              S4Vectors_0.40.2           
[59] BiocGenerics_0.48.1         corrplot_0.92              
[61] lubridate_1.9.3             forcats_1.0.0              
[63] purrr_1.0.2                 readr_2.1.5                
[65] tidyverse_2.0.0             stringr_1.5.1              
[67] tidyr_1.3.0                 scales_1.3.0               
[69] data.table_1.14.10          readxl_1.4.3               
[71] tibble_3.2.1                magrittr_2.0.3             
[73] reshape2_1.4.4              ggplot2_3.4.4              
[75] dplyr_1.1.4                 workflowr_1.7.1            

loaded via a namespace (and not attached):
  [1] fs_1.6.3                      ProtGenerics_1.34.0          
  [3] bitops_1.0-7                  httr_1.4.7                   
  [5] RColorBrewer_1.1-3            backports_1.4.1              
  [7] utf8_1.2.4                    R6_2.5.1                     
  [9] lazyeval_0.2.2                mgcv_1.9-1                   
 [11] withr_3.0.0                   prettyunits_1.2.0            
 [13] cli_3.6.2                     formatR_1.14                 
 [15] labeling_0.4.3                sass_0.4.8                   
 [17] Rsamtools_2.18.0              systemfonts_1.0.5            
 [19] rstudioapi_0.15.0             RSQLite_2.3.5                
 [21] generics_0.1.3                BiocIO_1.12.0                
 [23] crosstalk_1.2.1               car_3.1-2                    
 [25] zip_2.3.0                     GO.db_3.18.0                 
 [27] Matrix_1.6-5                  fansi_1.0.6                  
 [29] abind_1.4-5                   lifecycle_1.0.4              
 [31] whisker_0.4.1                 yaml_2.3.8                   
 [33] carData_3.0-5                 SparseArray_1.2.3            
 [35] blob_1.2.4                    promises_1.2.1               
 [37] crayon_1.5.2                  lattice_0.22-5               
 [39] KEGGREST_1.42.0               pillar_1.9.0                 
 [41] knitr_1.45                    rjson_0.2.21                 
 [43] admisc_0.34                   codetools_0.2-19             
 [45] fastmatch_1.1-4               getPass_0.2-4                
 [47] vctrs_0.6.5                   png_0.1-8                    
 [49] cellranger_1.1.0              gtable_0.3.4                 
 [51] cachem_1.0.8                  xfun_0.41                    
 [53] S4Arrays_1.2.0                mime_0.12                    
 [55] statmod_1.5.0                 interactiveDisplayBase_1.40.0
 [57] ellipsis_0.3.2                nlme_3.1-164                 
 [59] bit64_4.0.5                   progress_1.2.3               
 [61] filelock_1.0.3                rprojroot_2.0.4              
 [63] bslib_0.6.1                   colorspace_2.1-0             
 [65] DBI_1.2.1                     tidyselect_1.2.0             
 [67] processx_3.8.3                bit_4.0.5                    
 [69] compiler_4.3.3                curl_5.2.0                   
 [71] git2r_0.33.0                  xml2_1.3.6                   
 [73] ggdendro_0.1.23               DelayedArray_0.28.0          
 [75] plotly_4.10.4                 rtracklayer_1.62.0           
 [77] callr_3.7.3                   rappdirs_0.3.3               
 [79] digest_0.6.34                 rmarkdown_2.25               
 [81] XVector_0.42.0                htmltools_0.5.7              
 [83] pkgconfig_2.0.3               highr_0.10                   
 [85] fastmap_1.1.1                 rlang_1.1.3                  
 [87] htmlwidgets_1.6.4             shiny_1.8.0                  
 [89] farver_2.1.1                  jquerylib_0.1.4              
 [91] zoo_1.8-12                    jsonlite_1.8.8               
 [93] BiocParallel_1.36.0           RCurl_1.98-1.14              
 [95] GenomeInfoDbData_1.2.11       munsell_0.5.0                
 [97] Rcpp_1.0.12                   babelgene_22.9               
 [99] visdat_0.6.0                  stringi_1.8.3                
[101] zlibbioc_1.48.0               MASS_7.3-60.0.1              
[103] plyr_1.8.9                    parallel_4.3.3               
[105] Biostrings_2.70.1             splines_4.3.3                
[107] pander_0.6.5                  hms_1.1.3                    
[109] locfit_1.5-9.8                ps_1.7.6                     
[111] uuid_1.2-0                    ggsignif_0.6.4               
[113] futile.options_1.0.1          BiocVersion_3.18.1           
[115] evaluate_0.23                 lambda.r_1.2.4               
[117] BiocManager_1.30.22           tzdb_0.4.0                   
[119] httpuv_1.6.13                 xtable_1.8-4                 
[121] restfulr_0.0.15               rstatix_0.7.2                
[123] later_1.3.2                   memoise_2.0.1                
[125] GenomicAlignments_1.38.2      timechange_0.3.0             
[127] here_1.0.1                   </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->
<script>
$(document).ready(function () {
  window.initializeCodeFolding("hide" === "show");
});
</script>

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
