---
title: "DEG-analysis"
author: "urwahnawaz"
date: "2024-01-12"
output: 
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  chunk_output_type: console
---

## Introduction

```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```


```{r}
source(here::here("code/libraries.R"))
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(gridExtra)
library(EnsDb.Hsapiens.v86)
library(ggrepel)
library(org.Hs.eg.db)
library(msigdbr)
library(fgsea)

goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
    readRDS() %>%
    mutate(ontology = as.character(ontology))

getGeneLists <- function(pwf, goterms, genome, ids){
    gene2cat <- getgo(rownames(pwf), genome, ids)
    cat2gene <- split(rep(names(gene2cat), sapply(gene2cat, length)),
                      unlist(gene2cat, use.names = FALSE))
    out <- list()
    for(term in goterms){
        tmp <- pwf[cat2gene[[term]],]
        tmp <- rownames(tmp[tmp$DEgenes > 0, ])
        out[[term]] <- tmp
    }
    out


}
```


```{r}
txdf = transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")])
```



```{r}
ah <- AnnotationHub() %>%
    subset(species == "Homo sapiens") %>%
    subset(rdataclass == "EnsDb") %>% 
  subset(genome == "GRCh38")

ensDb <- ah[["AH109606"]]
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]

genesGR = genes(ensDb)
transGR = transcripts(ensDb)

mcols(transGR) = mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )


id2Name <- structure(
  genesGR$gene_name,
  names = genesGR$gene_id
) %>% 
  .[!duplicated(names(.))]

```



```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon/", "", salmon)
sample_names = gsub(".gz_transcripts", "", sample_names)
sample_names = gsub("\\_.*", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything()) %>% 
  mutate(names = gsub("\\_.*", "", names) )
md = md[order(match(md$names, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)

md %<>% dplyr::filter(Group != "FMR1") %>%
 dplyr::filter(names != "23-LDJ6767") %>%
dplyr::filter(names != "202")
all_files = all_files[names(all_files) %in% md$names]
```


```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= 1 ) >= 3)

```


# Differential gene expression analysis 

```{r}
txi_genes_filtered = txi_genes$counts[keep.genes,]
y <- DGEList(txi_genes_filtered)


design <- model.matrix(~Batch +Group + Sex, data = md) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

y <- calcNormFactors(y)
v <- voom(y, design)

fit = lmFit(v, design) %>% 
    eBayes()

```
 
```{r}
summary(decideTests(fit, lfc =0))
```

## UPF1 DEGs

```{r}
upf1_results_lfc = topTable(fit,coef = "UPF1", number = Inf) %>% 
                     mutate(res = ifelse(logFC > 0& adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0 & adj.P.Val < 0.05, "Downregulated", "NotSig"))) %>% 
  rownames_to_column("ensembl_gene_id") %>%
   mutate(SYMBOL = mapIds(org.Hs.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  mutate(res = ifelse(logFC > 0 & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0  & adj.P.Val < 0.05, "Downregulated", "NotSig")))
```

```{r fig.height=8, fig.width=10, fig.cap="Volcano plot showing distribution of differentially expressed genes using the limma/voom pipeline"}

DEColours <- c("Downregulated" = "#2e294e","Upregulated" = "#720026",  "NotSig" = "#E5E5E5")


volc_upf1 = upf1_results_lfc  %>%
    ggplot(aes(y = -log10(adj.P.Val), 
               x =  logFC , 
               colour = res,
               size =-log10(adj.P.Val), 
               label= SYMBOL)) +
  geom_point(alpha = 0.8) +
  # geom_text(aes(label=ifelse(SYMBOL== "Upf1",as.character(SYMBOL),''))) +
  #  geom_text(aes(label= SYMBOL), subset = SYMBOL == "Upf1") +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") +
  xlim(-8.5, 8.5) + ylim(0, 7.5)


# volc = volc_upf1+ geom_text_repel(data=subset(upf1_results_lfc , SYMBOL %in% c("UPF1", "UPF2", "UPF3B",
#                       "SMG5", "SMG6",
#                       "UPF3A", "ATF4", 
#                       "GADD5G")),
#             aes(label=SYMBOL),   position=position_dodge(width = 0.9), 
#              vjust=-0.40, color = "black", box.padding = 0.5, fill = "white") + ggtitle("DEGs in UPF1 relative to controls using limma/voom")
# 
# 
# volc 

my_gg = volc_upf1 + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```

```{r}
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  dplyr::arrange(adj.P.Val) %>% 
  DT::datatable(caption="DEGS in UPF1 relative to controls using limma/voom pipeline") 
```


<!-- # ```{r} -->
<!-- # sig_genes = upf1_results_lfc %>%  -->
<!-- #   dplyr::filter(adj.P.Val < 0.05) %>%  -->
<!-- #   dplyr::select(ensembl_gene_id)  -->
<!-- #  -->
<!-- # not_na = !is.na(genes) -->
<!-- # names(genes) = upf1_results_lfc$ensembl_gene_id -->
<!-- # genes = genes[not_na] -->
<!-- #      -->
<!-- # pwf=nullp(genes,plot.fit = FALSE, 'hg19','ensGene') -->
<!-- # Go.wall=goseq(pwf, 'hg19','ensGene', use_genes_without_cat = TRUE) -->
<!-- # goterms <- Go.wall$category -->
<!-- # goList <- getGeneLists(pwf, goterms, 'hg19','ensGene') -->
<!-- # Go.wall$EnsemblID <- sapply(Go.wall$category, function(x) goList[[x]]) -->
<!-- # Go.wall$EnsemblID <- vapply(Go.wall$EnsemblID, paste, collapse = ", ", character(1L)) -->
<!-- # ``` -->



## FRAX DEGs
```{r}
frax_results_lfc = topTable(fit,coef = "FRAX", number = Inf) %>% 
                     mutate(res = ifelse(logFC > 0  & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0 & adj.P.Val < 0.05, "Downregulated", "NotSig"))) %>% 
  rownames_to_column("ensembl_gene_id") %>%
   mutate(SYMBOL = mapIds(org.Hs.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
    mutate(res = ifelse(logFC > 0 & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0  & adj.P.Val < 0.05, "Downregulated", "NotSig"))) 
```

```{r}
save(frax_results_lfc,upf1_results_lfc, file = here::here("output/DEG-results.Rda"))
```


```{r}
DEColours <- c("Downregulated" = "#2e294e","Upregulated" = "#720026",  "NotSig" = "#E5E5E5")


volc_frax = frax_results_lfc  %>%
    ggplot(aes(y = -log10(adj.P.Val), 
               x =  logFC , 
               colour = res,
               size =-log10(adj.P.Val), 
               label= SYMBOL)) +
  geom_point(alpha = 0.8) +
  # geom_text(aes(label=ifelse(SYMBOL== "Upf1",as.character(SYMBOL),''))) +
  #  geom_text(aes(label= SYMBOL), subset = SYMBOL == "Upf1") +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") +
  xlim(-8.5, 8.5) + ylim(0, 7.5)


# volc = volc + geom_text_repel(data=subset(upf1_results_lfc , SYMBOL %in% c("UPF1", "UPF2", "UPF3B",
#                       "SMG5", "SMG6",
#                       "UPF3A", "ATF4", 
#                       "GADD5G", "FMR1")),
#             aes(label=SYMBOL),   position=position_dodge(width = 0.9), 
#              vjust=-0.40, color = "black", box.padding = 0.5, fill = "white") + ggtitle("DEGs in FRAX relative to controls using limma/voom pipeline")


my_gg =  volc_frax + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```





## Overlap of UPF1 and FRAX

* Number of genes overlapping
```{r}
library(VennDiagram)
x= list("UPF1 DEGs" = upf1_results_lfc$ensembl_gene_id[upf1_results_lfc$adj.P.Val < 0.05], 
        "FRAX DEGs" = frax_results_lfc$ensembl_gene_id[frax_results_lfc$adj.P.Val < 0.05])

ggvenn(x)
```

```{r}
overlaps = calculate.overlap(x)
upf1_only = overlaps$a1[!overlaps$a1 %in% overlaps$a3]
fmr1_only = overlaps$a2[!overlaps$a2 %in% overlaps$a3]
```

```{r fig.cap="Overlap of significant genes from FRAX and UPF1"}
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
               dplyr::filter(adj.P.Val < 0.05) %>%
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>% 
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 222 overlapping genes")

```

- Gene ontology

```{r}
genes = as.integer(upf1_results_lfc$ensembl_gene_id %in% overlaps$a3)
not_na = !is.na(genes)
names(genes) = upf1_results_lfc$ensembl_gene_id
genes = genes[not_na]
    
pwf=nullp(genes,plot.fit = FALSE, 'hg19','ensGene')
Go.wall=goseq(pwf, 'hg19','ensGene', use_genes_without_cat = TRUE)
goterms <- Go.wall$category
goList <- getGeneLists(pwf, goterms, 'hg19','ensGene')
Go.wall$EnsemblID <- sapply(Go.wall$category, function(x) goList[[x]])
Go.wall$EnsemblID <- vapply(Go.wall$EnsemblID, paste, collapse = ", ", character(1L))
```

```{r}
goRes = Go.wall %>%
            dplyr::select(-under_represented_pvalue) %>%
            dplyr::filter(numDEInCat > 0) %>%
            dplyr::rename(id = category) %>%
            left_join(goSummaries)
nGenes <- c(BG = sum(genes == 0), DE = sum(genes))
goFinal <- goRes %>%
  dplyr::filter(numDEInCat > 2) %>% 
  dplyr::filter(shortest_path > 3) %>%
  arrange(desc(shortest_path), EnsemblID) %>% 
  distinct(EnsemblID, .keep_all = TRUE) %>%
  arrange(over_represented_pvalue) %>% 
  mutate(Expected = round(nGenes[["DE"]] * numInCat / nGenes[["BG"]], 0)) %>%
            mutate(adjP = p.adjust(over_represented_pvalue, "bonferroni"),
                   FDR = p.adjust(over_represented_pvalue, "fdr"),
                   Sig_adjP = adjP < 0.05, Sig_FDR = FDR < 0.05) %>%
            dplyr::filter(Sig_adjP, numDEInCat > Expected) %>%
  dplyr::select(term, everything()) 
  
  
goFinal %>% 
  DT::datatable(caption="222 overlapping genes")

```


- UPF1 specific genes 

```{r}
upf1_results_lfc %>% 
  dplyr::filter(ensembl_gene_id %in% upf1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>%
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 441 UPF1 specific genes")

```

```{r}
genes = as.integer(upf1_results_lfc$ensembl_gene_id %in% upf1_only)
not_na = !is.na(genes)
names(genes) = upf1_results_lfc$ensembl_gene_id
genes = genes[not_na]
    
pwf=nullp(genes,plot.fit = FALSE, 'hg19','ensGene')
Go.wall=goseq(pwf, 'hg19','ensGene', use_genes_without_cat = TRUE)
goterms <- Go.wall$category
goList <- getGeneLists(pwf, goterms, 'hg19','ensGene')
Go.wall$EnsemblID <- sapply(Go.wall$category, function(x) goList[[x]])
Go.wall$EnsemblID <- vapply(Go.wall$EnsemblID, paste, collapse = ", ", character(1L))
```

```{r}
goRes = Go.wall %>%
            dplyr::select(-under_represented_pvalue) %>%
            dplyr::filter(numDEInCat > 0) %>%
            dplyr::rename(id = category) %>%
            left_join(goSummaries)
nGenes <- c(BG = sum(genes == 0), DE = sum(genes))
goFinal <- goRes %>%
  dplyr::filter(numDEInCat > 2) %>% 
  dplyr::filter(shortest_path > 4) %>%
  arrange(desc(shortest_path), EnsemblID) %>% 
  distinct(EnsemblID, .keep_all = TRUE) %>%
  arrange(over_represented_pvalue) %>% 
  mutate(Expected = round(nGenes[["DE"]] * numInCat / nGenes[["BG"]], 0)) %>%
            mutate(adjP = p.adjust(over_represented_pvalue, "bonferroni"),
                   FDR = p.adjust(over_represented_pvalue, "fdr"),
                   Sig_adjP = adjP < 0.05, Sig_FDR = FDR < 0.05) %>%
            dplyr::filter(Sig_adjP, numDEInCat > Expected) %>%
  dplyr::select(term, everything()) 
  
  
goFinal %>% 
  DT::datatable(caption="UPF1 specific genes")

```


- FMR1 specific genes 

```{r}
frax_results_lfc %>% 
  dplyr::filter(ensembl_gene_id %in% fmr1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val) %>% 
  inner_join(upf1_results_lfc %>% 
                dplyr::select(ensembl_gene_id, logFC, 
                            adj.P.Val), by = "ensembl_gene_id") %>%
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 1178 FRAX specific genes")

```


Gene ontology


```{r}
genes = as.integer(frax_results_lfc$ensembl_gene_id %in% fmr1_only)
not_na = !is.na(genes)
names(genes) = frax_results_lfc$ensembl_gene_id
genes = genes[not_na]
    
pwf=nullp(genes,plot.fit = FALSE, 'hg19','ensGene')
Go.wall=goseq(pwf, 'hg19','ensGene', use_genes_without_cat = TRUE)
goterms <- Go.wall$category
goList <- getGeneLists(pwf, goterms, 'hg19','ensGene')
Go.wall$EnsemblID <- sapply(Go.wall$category, function(x) goList[[x]])
Go.wall$EnsemblID <- vapply(Go.wall$EnsemblID, paste, collapse = ", ", character(1L))
```

```{r}
goRes = Go.wall %>%
            dplyr::select(-under_represented_pvalue) %>%
            dplyr::filter(numDEInCat > 0) %>%
            dplyr::rename(id = category) %>%
            left_join(goSummaries)
nGenes <- c(BG = sum(genes == 0), DE = sum(genes))
goFinal <- goRes %>%
  dplyr::filter(numDEInCat > 2) %>% 
  dplyr::filter(shortest_path > 4) %>%
  arrange(desc(shortest_path), EnsemblID) %>% 
  distinct(EnsemblID, .keep_all = TRUE) %>%
  arrange(over_represented_pvalue) %>% 
  mutate(Expected = round(nGenes[["DE"]] * numInCat / nGenes[["BG"]], 0)) %>%
            mutate(adjP = p.adjust(over_represented_pvalue, "bonferroni"),
                   FDR = p.adjust(over_represented_pvalue, "fdr"),
                   Sig_adjP = adjP < 0.05, Sig_FDR = FDR < 0.05) %>%
            dplyr::filter(Sig_adjP, numDEInCat > Expected) %>%
  dplyr::select(term, everything()) 
  
  
goFinal %>% 
  DT::datatable(caption="FRAX specific genes")

```


# Comparison of DEGs with DSB genes from Chakraborty et al

```{r}
dsbs = read_xlsx(here::here("data/lists/DSBs/FX_DSB_genes.xlsx"), skip = 1)
exp_cl = read_xlsx(here::here("data/lists/DSBs/msrd_results_LCL_Silent_Genes.xlsx"), sheet = 1)

dsbs_lcls = dsbs[!dsbs$ensembl_gene_id %in% exp_cl$ensembl_id,]
```

* Jozef's questions 
* Is there an overlap of genes with DSBs in FRAXA/FMR1 LCLs and the DEGs in FRAXA/FMR1 LCLs even if across 2x different studies? If not overlap based on p value, is there a trend for these DSB-associated genes to be downregulated?
*Is there an overlap of genes with DSBs in FRAXA/FMR1 LCLs and DEGs in UPF3B patient LCLs, the reason being that UPF3B patients have no or very little FMRP left as Saba determined? Also, if no overlap strictly on p value, is there a trend of the DSB genes to be down-regulated in UPF3B LCLs?
* The same as above, but comparing the DSB genes to UPF1 DEGs, etc.


- From this study - using the FRAX DEGs and UPF1 DEGs 

```{r}
x= list("UPF1 DEGs" = upf1_results_lfc$ensembl_gene_id[upf1_results_lfc$adj.P.Val < 0.05], 
        "FRAX DEGs" = frax_results_lfc$ensembl_gene_id[frax_results_lfc$adj.P.Val < 0.05], 
        "Double stranded breaks" = dsbs$ensembl_gene_id)

ggvenn(x)

```


```{r fig.cap="Genes with DSBs identified in significantly expressed DEGs"}
frax_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  mutate(condition="FRAX") %>% 
  inner_join(dsbs, by = "ensembl_gene_id") %>% 
  rbind(upf1_results_lfc %>% 
          dplyr::filter(adj.P.Val < 0.05) %>%
          mutate(condition="UPF1") %>% 
  inner_join(dsbs, by = "ensembl_gene_id")) %>%
  ggplot(aes(x=condition, fill = res), color = "white") + geom_bar(stat = "count") + 
  theme_bw() + scale_fill_manual(values =c("Upregulated"=  "#720026", 
                                           "Downregulated"= "#2e294e")) + 
  ggtitle("Number of DEGs with DSBs")
```

```{r fig.cap="Genes with DSBs identified in significantly expressed DEGs"}
library(ggstatsplot)
frax_results_lfc %>% 
  mutate(res = ifelse(logFC > 0, "Upregulated", 
                      ifelse(logFC < 0, "Downregulated", "NotSig"))) %>%
  mutate(condition="FRAX") %>% 
  inner_join(dsbs, by = "ensembl_gene_id") %>% 
  rbind(upf1_results_lfc %>% 
         mutate(res = ifelse(logFC > 0, "Upregulated", 
                      ifelse(logFC < 0, "Downregulated", "NotSig"))) %>%
          mutate(condition="UPF1") %>% 
  inner_join(dsbs, by = "ensembl_gene_id")) %>%
  ggbarstats(
  x                = res,
  y                = condition,

xlab = "",
legend.title     = "",
  ggplot.component = list(ggplot2::scale_x_discrete(guide = ggplot2::guide_axis(n.dodge = 2))),
) + scale_fill_manual(values =c("Upregulated"=  "#720026", 
                                           "Downregulated"= "#2e294e"))


  
 # ggplot(aes(x=condition, fill = res), color = "white") + geom_bar(stat = "count") + 
  #theme_bw() + scale_fill_manual(values =c("Upregulated"=  "#720026", 
             #                              "Downregulated"= "#2e294e")) + 
  #ggtitle("Number of DEGs with DSBs")
```

```{r}
library(GeneOverlap)
enrichment_gene_list_pvalue = list()
enrichment_gene_list_OR = list()

test.list = list()
test.list$BG = frax_results_lfc$ensembl_gene_id
test.list$frax_sig = frax_results_lfc$ensembl_gene_id[frax_results_lfc$adj.P.Val < 0.05]
test.list$frax_sig_up = frax_results_lfc$ensembl_gene_id[frax_results_lfc$adj.P.Val < 0.05 & frax_results_lfc$logFC > 0]
test.list$frax_sig_down = frax_results_lfc$ensembl_gene_id[frax_results_lfc$adj.P.Val < 0.05 & frax_results_lfc$logFC < 0]
test.list$upf1_sig = upf1_results_lfc$ensembl_gene_id[upf1_results_lfc$adj.P.Val < 0.05]
test.list$upf1_sig_up = upf1_results_lfc$ensembl_gene_id[upf1_results_lfc$adj.P.Val < 0.05 & upf1_results_lfc$logFC > 0]
test.list$upf1_sig_down = upf1_results_lfc$ensembl_gene_id[upf1_results_lfc$adj.P.Val < 0.05 & upf1_results_lfc$logFC < 0]
test.list$dsbs = dsbs$ensembl_gene_id
## Maybe an exact test with odds ratio 

for (list in c("frax_sig","frax_sig_up","frax_sig_down","upf1_sig",
               "upf1_sig_up", "upf1_sig_down")) {
    go.obj <- newGeneOverlap(test.list[[list]],
                         test.list$dsbs,
                         genome.size=length(test.list$BG))
    go.obj <- testGeneOverlap(go.obj)
    enrichment_gene_list_pvalue[[paste0(list)]] = go.obj@pval
    enrichment_gene_list_OR[[paste0(list)]] = go.obj@odds.ratio
    
  }


enrichment_GL_pval = do.call(rbind, enrichment_gene_list_pvalue) %>% as.data.frame()
colnames(enrichment_GL_pval)[1] = c("pvalue")
enrichment_GL_pval$padj = p.adjust(enrichment_GL_pval$pvalue, method = "BH")
enrichment_GL_OR  = do.call(rbind, enrichment_gene_list_OR) %>% as.data.frame()
colnames(enrichment_GL_OR)[1] = c("OR")
GL_enrichment = cbind(enrichment_GL_pval, enrichment_GL_OR)
```

```{r}
GL_enrichment %>% 
  rownames_to_column("DEGs") %>%
  ggplot(aes(x=OR, y = DEGs)) + geom_bar(stat = "identity", fill = "#edafb8") +
  theme_bw() + geom_vline(xintercept = 1) + 
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=12, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(x="Odds ratio") + ggtitle("Enrichment of DSBs in UPF1 or FRAX")

```


### List of genes with DSBs

```{r}
frax_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  mutate(condition="FRAX") %>% 
  inner_join(dsbs, by = "ensembl_gene_id") %>%
  dplyr::select(gene = SYMBOL, Expression = res, everything()) %>%
  DT::datatable(caption = "Genes in FRAX DEGs with DSBs")
```


```{r}
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  mutate(condition="FRAX") %>% 
  inner_join(dsbs, by = "ensembl_gene_id") %>%
  dplyr::select(gene = SYMBOL, Expression = res, everything()) %>%
  DT::datatable(caption = "Genes in UPF1 DEGs with DSBs")
```

### Expression overlap with genes with DSBs 

```{r}
library(ggside)
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
               dplyr::filter(adj.P.Val < 0.05) %>%
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>% 
  mutate(DSBs = ifelse(ensembl_gene_id %in% dsbs$ensembl_gene_id, "DSBs", "No DSBs" )) %>%
 ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line",
              conf.int = TRUE, color = "DSBs", add.params = list(color = "blue",
                                                 fill = "lightgray"), alpha =0.8, shape = "DSBs", size =3) + 
    theme_bw() + xlab("UPF1 (log2FC)") +
    scale_color_manual(values = c("#F15A2B", "#2E368F")) + 
    geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(color = "Expression") + geom_xsideboxplot(aes(y = DSBs, fill= DSBs), orientation = "y") +
   theme(ggside.panel.scale = .3) +
  scale_xsidey_discrete()  + geom_ysidedensity(aes(x = after_stat(density), fill = DSBs), alpha = 0.5) +
  scale_color_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) +  scale_fill_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) + scale_shape_manual(values = c(19,15)) + labs(y= "FRAX (log2FC)")
```

```{r}
upf1_results_lfc %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc  %>%
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>% 
  mutate(DSBs = ifelse(ensembl_gene_id %in% dsbs$ensembl_gene_id, "DSBs", "No DSBs" )) %>%
 ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line",
              conf.int = TRUE, color = "DSBs", add.params = list(color = "blue",
                                                 fill = "lightgray"), alpha =0.8, shape = "DSBs", size =3) + 
    theme_bw() + xlab("UPF1 (log2FC)") +
    scale_color_manual(values = c("#F15A2B", "#2E368F")) + 
    geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(color = "Expression") + geom_xsideboxplot(aes(y = DSBs, fill= DSBs), orientation = "y") +
   theme(ggside.panel.scale = .3) +
  scale_xsidey_discrete()  + geom_ysidedensity(aes(x = after_stat(density), fill = DSBs), alpha = 0.5) +
  scale_color_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) +  scale_fill_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) + scale_shape_manual(values = c(19,15)) + labs(y= "FRAX (log2FC)")

```

```{r}
frax_results_lfc %>% 
  dplyr::filter(ensembl_gene_id %in% fmr1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val) %>% 
  inner_join(upf1_results_lfc %>% 
                dplyr::select(ensembl_gene_id, logFC, 
                            adj.P.Val), by = "ensembl_gene_id") %>% 
  mutate(DSBs = ifelse(ensembl_gene_id %in% dsbs$ensembl_gene_id, "DSBs", "No DSBs" )) %>%
 ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line",
              conf.int = TRUE, color = "DSBs", add.params = list(color = "blue",
                                                 fill = "lightgray"), alpha =0.8, shape = "DSBs", size =3) + 
    theme_bw() + xlab("UPF1 (log2FC)") +
    scale_color_manual(values = c("#F15A2B", "#2E368F")) + 
    geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(color = "Expression") + geom_xsideboxplot(aes(y = DSBs, fill= DSBs), orientation = "y") +
   theme(ggside.panel.scale = .3) +
  scale_xsidey_discrete()  + geom_ysidedensity(aes(x = after_stat(density), fill = DSBs), alpha = 0.5) +
  scale_color_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) +  scale_fill_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) + scale_shape_manual(values = c(19,15)) + labs(y= "FRAX (log2FC)")
```

```{r}
upf1_results_lfc %>% 
  dplyr::filter(ensembl_gene_id %in% upf1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>% 
  mutate(DSBs = ifelse(ensembl_gene_id %in% dsbs$ensembl_gene_id, "DSBs", "No DSBs" )) %>%
 ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line",
              conf.int = TRUE, color = "DSBs", add.params = list(color = "blue",
                                                 fill = "lightgray"), alpha =0.8, shape = "DSBs", size =3) + 
    theme_bw() + xlab("UPF1 (log2FC)") +
    scale_color_manual(values = c("#F15A2B", "#2E368F")) + 
    geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + theme(legend.position = "none") +
  theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=15, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(color = "Expression") + geom_xsideboxplot(aes(y = DSBs, fill= DSBs), orientation = "y") +
   theme(ggside.panel.scale = .3) +
  scale_xsidey_discrete()  + geom_ysidedensity(aes(x = after_stat(density), fill = DSBs), alpha = 0.5) +
  scale_color_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) +  scale_fill_manual(values = c("DSBs" = "#b23a48",
                                "No DSBs" = "grey")) + scale_shape_manual(values = c(19,15)) + labs(y= "FRAX (log2FC)")
```


# Data export

