---
title: "DEG-analysis"
author: "urwahnawaz"
date: "2024-01-12"
output: 
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  chunk_output_type: console
---

## Introduction

```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```


```{r}
source(here::here("code/libraries.R"))
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(gridExtra)
library(EnsDb.Hsapiens.v86)
library(ggrepel)
library(org.Hs.eg.db)
library(msigdbr)
library(fgsea)

goSummaries <- url("https://uofabioinformaticshub.github.io/summaries2GO/data/goSummaries.RDS") %>%
    readRDS() %>%
    mutate(ontology = as.character(ontology))

getGeneLists <- function(pwf, goterms, genome, ids){
    gene2cat <- getgo(rownames(pwf), genome, ids)
    cat2gene <- split(rep(names(gene2cat), sapply(gene2cat, length)),
                      unlist(gene2cat, use.names = FALSE))
    out <- list()
    for(term in goterms){
        tmp <- pwf[cat2gene[[term]],]
        tmp <- rownames(tmp[tmp$DEgenes > 0, ])
        out[[term]] <- tmp
    }
    out


}
```


```{r}
txdf = ensembldb::transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")])

gene2symbol = ensembldb::genes(EnsDb.Hsapiens.v86, return.type="DataFrame") %>% 
  as.data.frame() %>% 
  dplyr::select(gene_id, gene_name)
```



```{r}
ah <- AnnotationHub() %>%
    subset(species == "Homo sapiens") %>%
    subset(rdataclass == "EnsDb") %>% 
  subset(genome == "GRCh38")

ensDb <- ah[["AH109606"]]
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]

genesGR = ensembldb::genes(ensDb)
transGR = transcripts(ensDb)

mcols(transGR) = mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )


id2Name <- structure(
  genesGR$gene_name,
  names = genesGR$gene_id
) %>% 
  .[!duplicated(names(.))]

```



```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon/", "", salmon)
sample_names = gsub(".gz_transcripts", "", sample_names)
sample_names = gsub("\\_.*", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything()) %>% 
  mutate(names = gsub("\\_.*", "", names) )
md = md[order(match(md$names, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)

md %<>% dplyr::filter(Group != "FMR1") %>%
 dplyr::filter(names != "23-LDJ6767") %>%
dplyr::filter(names != "202")
all_files = all_files[names(all_files) %in% md$names]
```


```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= 1 ) >= 3)

```


# Differential gene expression analysis 

```{r}
txi_genes_filtered = txi_genes$counts[keep.genes,]
y <- DGEList(txi_genes_filtered)


design <- model.matrix(~Batch +Group + Sex, data = md) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

y <- calcNormFactors(y)
v <- voom(y, design)

fit = lmFit(v, design) %>% 
    eBayes()

```
 
```{r}
summary(decideTests(fit, lfc =0))
```

## UPF1 DEGs

```{r}
upf1_results_lfc = topTable(fit,coef = "UPF1", number = Inf) %>% 
                     mutate(res = ifelse(logFC > 0& adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0 & adj.P.Val < 0.05, "Downregulated", "NotSig"))) %>% 
  rownames_to_column("ensembl_gene_id") %>%
   mutate(SYMBOL = mapIds(org.Hs.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>%
  mutate(res = ifelse(logFC > 0 & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0  & adj.P.Val < 0.05, "Downregulated", "NotSig")))
```

```{r fig.height=8, fig.width=10, fig.cap="Volcano plot showing distribution of differentially expressed genes using the limma/voom pipeline"}

DEColours <- c("Downregulated" = "#2e294e","Upregulated" = "#720026",  "NotSig" = "#E5E5E5")


volc_upf1 = upf1_results_lfc  %>%
    ggplot(aes(y = -log10(adj.P.Val), 
               x =  logFC , 
               colour = res,
               size =-log10(adj.P.Val), 
               label= SYMBOL)) +
  geom_point(alpha = 0.8) +
  # geom_text(aes(label=ifelse(SYMBOL== "Upf1",as.character(SYMBOL),''))) +
  #  geom_text(aes(label= SYMBOL), subset = SYMBOL == "Upf1") +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") +
  xlim(-8.5, 8.5) + ylim(0, 7.5)


# volc = volc_upf1+ geom_text_repel(data=subset(upf1_results_lfc , SYMBOL %in% c("UPF1", "UPF2", "UPF3B",
#                       "SMG5", "SMG6",
#                       "UPF3A", "ATF4", 
#                       "GADD5G")),
#             aes(label=SYMBOL),   position=position_dodge(width = 0.9), 
#              vjust=-0.40, color = "black", box.padding = 0.5, fill = "white") + ggtitle("DEGs in UPF1 relative to controls using limma/voom")
# 
# 
# volc 

my_gg = volc_upf1 + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```

```{r}
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  dplyr::arrange(adj.P.Val) %>% 
  DT::datatable(caption="DEGS in UPF1 relative to controls using limma/voom pipeline") 
```


- Gene ontology enrichment analysis 

### Gene tonic for visualisation
```{r}
library(GeneTonic)
library(DESeq2)
library(AnnotationDbi)
library(pcaExplorer)
library(igraph)
library(visNetwork)
library(magrittr)
library(topGO)


```


```{r}
tbledger <- as.data.frame(upf1_results_lfc)
colnames(tbledger)[colnames(tbledger) == "P.Value"] <- "pvalue"
colnames(tbledger)[colnames(tbledger) == "logFC"] <- "log2FoldChange"
colnames(tbledger)[colnames(tbledger) == "AveExpr"] <- "baseMean"
colnames(tbledger)[colnames(tbledger) == "SYMBOL"] <- "SYMBOL"
rownames(tbledger) =tbledger$ensembl_gene_id
tbledger$baseMean <- (2^tbledger$baseMean) * mean(y$samples$lib.size) / 1e6

edger_resu <- DESeqResults(DataFrame(tbledger))
edger_resu <- DESeq2:::pvalueAdjustment(edger_resu,
                                        independentFiltering = FALSE,
                                        alpha = FDR, pAdjustMethod = "BH")
```

```{r}
anno_df <- data.frame(
  gene_id = upf1_results_lfc$ensembl_gene_id,
  gene_name = mapIds(org.Hs.eg.db, keys = upf1_results_lfc$ensembl_gene_id, column = "SYMBOL", keytype = "ENSEMBL"),
  stringsAsFactors = FALSE,
  row.names = upf1_results_lfc$ensembl_gene_id
)

```

```{r}
sig_upf1 =upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) 

topgo_all_upf1_bp <-
  pcaExplorer::topGOtable(sig_upf1$SYMBOL,
                          upf1_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_upf1_bp <- shake_topGOtableResult(topgo_all_upf1_bp)

topgo_all_upf1_bp <- get_aggrscores(res_enrich = topgo_all_upf1_bp ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}
em <- enrichment_map(topgo_all_upf1_bp,
                     edger_resu,
                     n_gs = 30,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```



```{r}
library(plotly)
p <- enhance_table(topgo_all_upf1_bp,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_upf1_bp,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```

- Upregulated genes 

```{r}
sig_upf1 =upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05 & logFC > 0 ) 

topgo_all_upf1_bp <-
  pcaExplorer::topGOtable(sig_upf1$SYMBOL,
                          upf1_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500, 
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)

topgo_all_upf1_bp <- shake_topGOtableResult(topgo_all_upf1_bp)

topgo_all_upf1_bp  <- get_aggrscores(res_enrich = topgo_all_upf1_bp ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}

em <- enrichment_map(topgo_all_upf1_bp,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_upf1_bp,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_upf1_bp,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```


- Downregulated genes 

```{r}
sig_upf1 =upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05 & logFC < 0 ) 

topgo_all_upf1_bp <-
  pcaExplorer::topGOtable(sig_upf1$SYMBOL,
                          upf1_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500, 
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)

topgo_all_upf1_bp <- shake_topGOtableResult(topgo_all_upf1_bp)

topgo_all_upf1_bp  <- get_aggrscores(res_enrich = topgo_all_upf1_bp ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}

em <- enrichment_map(topgo_all_upf1_bp,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_upf1_bp,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_upf1_bp,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```

## FRAX DEGs
```{r}
frax_results_lfc = topTable(fit,coef = "FRAX", number = Inf) %>% 
                     mutate(res = ifelse(logFC > 0  & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0 & adj.P.Val < 0.05, "Downregulated", "NotSig"))) %>% 
  rownames_to_column("ensembl_gene_id") %>%
   mutate(SYMBOL = mapIds(org.Hs.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first")) %>% 
    mutate(res = ifelse(logFC > 0 & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < 0  & adj.P.Val < 0.05, "Downregulated", "NotSig"))) 
```

```{r}
save(frax_results_lfc,upf1_results_lfc, file = here::here("output/DEG-results.Rda"))
```


```{r}
DEColours <- c("Downregulated" = "#2e294e","Upregulated" = "#720026",  "NotSig" = "#E5E5E5")


volc_frax = frax_results_lfc  %>%
    ggplot(aes(y = -log10(adj.P.Val), 
               x =  logFC , 
               colour = res,
               size =-log10(adj.P.Val), 
               label= SYMBOL)) +
  geom_point(alpha = 0.8) +
  # geom_text(aes(label=ifelse(SYMBOL== "Upf1",as.character(SYMBOL),''))) +
  #  geom_text(aes(label= SYMBOL), subset = SYMBOL == "Upf1") +
    scale_colour_manual(values = DEColours) + theme_classic() + 
    theme(axis.title.y = element_text(size = 12)) +
    geom_hline(yintercept = -log10(0.05), color = "grey60", size = 0.5, lty = "dashed") +
    labs(x = "log2 Fold Change", y = "-log10 adj p-value") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") +
  xlim(-8.5, 8.5) + ylim(0, 7.5)


# volc = volc + geom_text_repel(data=subset(upf1_results_lfc , SYMBOL %in% c("UPF1", "UPF2", "UPF3B",
#                       "SMG5", "SMG6",
#                       "UPF3A", "ATF4", 
#                       "GADD5G", "FMR1")),
#             aes(label=SYMBOL),   position=position_dodge(width = 0.9), 
#              vjust=-0.40, color = "black", box.padding = 0.5, fill = "white") + ggtitle("DEGs in FRAX relative to controls using limma/voom pipeline")


my_gg =  volc_frax + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```

## Enrichment analysis 

```{r}
tbledger <- as.data.frame(frax_results_lfc)
colnames(tbledger)[colnames(tbledger) == "P.Value"] <- "pvalue"
colnames(tbledger)[colnames(tbledger) == "logFC"] <- "log2FoldChange"
colnames(tbledger)[colnames(tbledger) == "AveExpr"] <- "baseMean"
colnames(tbledger)[colnames(tbledger) == "SYMBOL"] <- "SYMBOL"
rownames(tbledger) =tbledger$ensembl_gene_id
tbledger$baseMean <- (2^tbledger$baseMean) * mean(y$samples$lib.size) / 1e6

edger_resu <- DESeqResults(DataFrame(tbledger))
edger_resu <- DESeq2:::pvalueAdjustment(edger_resu,
                                        independentFiltering = FALSE,
                                        alpha = FDR, pAdjustMethod = "BH")
```

```{r}
anno_df <- data.frame(
  gene_id = frax_results_lfc$ensembl_gene_id,
  gene_name = mapIds(org.Hs.eg.db, keys = frax_results_lfc$ensembl_gene_id, column = "SYMBOL", keytype = "ENSEMBL"),
  stringsAsFactors = FALSE,
  row.names = frax_results_lfc$ensembl_gene_id
)

```

```{r}
sig_frax=frax_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) 

topgo_all_frax <-
  pcaExplorer::topGOtable(sig_frax$SYMBOL,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}
em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 30,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```



```{r}
library(plotly)
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```

- Upregulated genes 

```{r}
sig_frax=frax_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05 & logFC > 0 ) 

topgo_all_frax <-
  pcaExplorer::topGOtable(sig_frax$SYMBOL,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}

em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```


- Downregulated genes 

```{r}
sig_frax=frax_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05 & logFC < 0 ) 

topgo_all_frax <-
  pcaExplorer::topGOtable(sig_frax$SYMBOL,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}

em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```


## Overlap of UPF1 and FRAX

* Number of genes overlapping
```{r}
library(VennDiagram)
x= list("UPF1 DEGs" = upf1_results_lfc$SYMBOL[upf1_results_lfc$adj.P.Val < 0.05], 
        "FRAX DEGs" = frax_results_lfc$SYMBOL[frax_results_lfc$adj.P.Val < 0.05])

ggvenn(x)
```

```{r}
overlaps = calculate.overlap(x)
upf1_only = overlaps$a1[!overlaps$a1 %in% overlaps$a3]
fmr1_only = overlaps$a2[!overlaps$a2 %in% overlaps$a3]
```

```{r fig.cap="Overlap of significant genes from FRAX and UPF1"}
upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
               dplyr::filter(adj.P.Val < 0.05) %>%
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>% 
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 222 overlapping genes")

```

- Gene ontology of overlapping genes 


```{r}
sig_genes =overlaps$a3

topgo_all_frax <-
  pcaExplorer::topGOtable(sig_genes,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```



```{r}

em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)

graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```








- UPF1 specific genes 

```{r}
upf1_results_lfc %>% 
  dplyr::filter(SYMBOL %in% upf1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(frax_results_lfc %>% 
                dplyr::select(ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val), by = "ensembl_gene_id") %>%
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 441 UPF1 specific genes")

```


```{r}
topgo_all_frax <-
  pcaExplorer::topGOtable(upf1_only,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```

```{r}
em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)
graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```

- FRAX specific genes 

```{r}
frax_results_lfc %>% 
  dplyr::filter(SYMBOL %in% fmr1_only) %>% 
  dplyr::select(SYMBOL,ensembl_gene_id, FRAX.logFC=logFC, 
                              FRAX.padj =adj.P.Val) %>% 
  inner_join(upf1_results_lfc %>% 
                dplyr::select(ensembl_gene_id, logFC, 
                            adj.P.Val), by = "ensembl_gene_id") %>%
  ggscatter(., x="logFC", y = "FRAX.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("FRAX (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 1178 FRAX specific genes")

```


```{r}
topgo_all_frax <-
  pcaExplorer::topGOtable(fmr1_only,
                         frax_results_lfc$SYMBOL,
                          ontology = "BP",
                          mapping = "org.Hs.eg.db",
                          geneID = "symbol",
                          topTablerows = 500,
                          do_padj = TRUE) %>%
  as.data.frame() %>%
  mutate(Ont = "BP") %>%
  dplyr::filter(p.value_classic < 0.05)


topgo_all_frax<- shake_topGOtableResult(topgo_all_frax)

topgo_all_frax<- get_aggrscores(res_enrich = topgo_all_frax ,
                                        res_de = edger_resu,
                                        annotation_obj = anno_df,
                                        aggrfun = mean)
```

```{r}
em <- enrichment_map(topgo_all_frax,
                     edger_resu,
                     n_gs = 50,
                     color_by = "z_score",
                     anno_df)
graph_vis = em %>% 
  visIgraph() %>% 
  visOptions(highlightNearest = list(enabled = TRUE,
                                     degree = 1,
                                     hover = TRUE),
             nodesIdSelection = TRUE)

graph_vis 
```


```{r}
p <- enhance_table(topgo_all_frax,
                   edger_resu,
                   n_gs = 30,
                   annotation_obj = anno_df,
                   chars_limit = 60)
ggplotly(p)
```


```{r}
distilled <- distill_enrichment(topgo_all_frax,
                                edger_resu,
                                anno_df,
                                n_gs = Inf,
                                cluster_fun = "cluster_markov")

DT::datatable(distilled$res_enrich[,])
```


## Enrichment analysis with other NMD LCLs 

```{r}
lists = list()
lists$upf3b = read_excel(here::here("data/lists/LCL-gene-list/UPF3B-variants.xlsx")) %>% as.data.frame() %>% 
  dplyr::rename("gene_name"= "external_gene_name") %>%
  left_join(gene2symbol, by = "gene_name") %>% 
  dplyr::select(log2FoldChan, padj, "EnsID" = gene_id , "symbol" = gene_name)


lists$upf2_group = read_excel(here::here("data/lists/LCL-gene-list/UPF2-CNV-grouped.xlsx")) %>% 
  as.data.frame() %>% 
  dplyr::select(log2FoldChang, padj, EnsID, "symbol" = external_gene_name)

lists$upf2_frameshift= read_excel(here::here("data/lists/LCL-gene-list/UPF2-frameshift-case.xlsx")) %>% as.data.frame() %>% 
  dplyr::select(log2FoldChange, padj, EnsID , "symbol" = external_gene_name)

lists$smg = read_excel(here::here("data/lists/LCL-gene-list/SMG-LCLs.xlsx")) %>% as.data.frame() %>%
  mutate_at(.vars = "EnsID", .funs = gsub, pattern = "\\.[0-9]*$", replacement = "")  %>% 
  dplyr::select(log2FoldChange, padj, EnsID, symbol)
```

Over-representation test 

### Transcript feature analysis 

```{r}
library(GeneOverlap)
overlaps_sig = lapply(lists, function(x){
  sig = x %>% 
    dplyr::filter(padj < 0.05)
  bg = calculate.overlap( x = list("x" = x$EnsID,
                         "upf1" = upf1_results_lfc$ensembl_gene_id))
  bg = bg$a3
  upf1_sig = upf1_results_lfc %>%
    dplyr::filter(adj.P.Val < 0.05)
    go.obj <- newGeneOverlap(sig$EnsID,
                         upf1_sig$ensembl_gene_id,
                         genome.size=length(upf1_results_lfc$ensembl_gene_id))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_pval= go.obj@pval
     enrichment_nif_OR= go.obj@odds.ratio
      NIF_feature_enrichment = cbind(enrichment_nif_pval, enrichment_nif_OR)
      return(NIF_feature_enrichment)
})

overlap_sig_frax = lapply(lists, function(x){
  sig = x %>% 
    dplyr::filter(padj < 0.05)
  bg = calculate.overlap( x = list("x" = x$EnsID,
                         "frax" = frax_results_lfc$ensembl_gene_id))
  bg = bg$a3
  frax_sig = frax_results_lfc %>%
    dplyr::filter(adj.P.Val < 0.05)
    go.obj <- newGeneOverlap(sig$EnsID,
                         frax_sig$ensembl_gene_id,
                         genome.size=length(frax_results_lfc$ensembl_gene_id))
     go.obj <- testGeneOverlap(go.obj)
     enrichment_nif_pval= go.obj@pval
     enrichment_nif_OR= go.obj@odds.ratio
      NIF_feature_enrichment = cbind(enrichment_nif_pval, enrichment_nif_OR)
      return(NIF_feature_enrichment)
})

```

```{r}
frax_sig = frax_results_lfc %>%
    dplyr::filter(adj.P.Val < 0.05)

upf1_sig = upf1_results_lfc %>%
    dplyr::filter(adj.P.Val < 0.05)
go.obj <- newGeneOverlap(upf1_sig$ensembl_gene_id,
                         frax_sig$ensembl_gene_id,
                         genome.size=length(upf1_results_lfc$ensembl_gene_id))
go.obj <- testGeneOverlap(go.obj)
enrichment_nif_pval= go.obj@pval
enrichment_nif_OR= go.obj@odds.ratio
NIF_feature_enrichment = cbind(enrichment_nif_pval, enrichment_nif_OR)

overlaps_sig$frax = NIF_feature_enrichment
```

## Correlation with UPF1 DEGs 
```{r}
do.call(rbind, overlaps_sig) %>% 
  as.data.frame() %>%
  mutate(overlap = c("UPF3B LCL DEGs", 
                     "UPF2 CNVs LCL DEGs", 
                     "UPF2 Frameshift LCLs", 
                     "SMG8/9 LCLs DEGs", 
                     "FRAX LCLs DEGs")) %>%
ggplot(aes(x=enrichment_nif_OR, y = overlap)) + geom_bar(stat = "identity", fill = "#edafb8") +
  theme_bw() + geom_vline(xintercept = 1) + 
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=12, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(x="Odds ratio", 
                  y = "") + 
  annotate("label", x =0.5, y = 5, label = "8.38-02") +
   annotate("label", x =0.5, y = 4, label = "4.6e-03") +
  annotate("label", x =0.5, y = 3, label = " 6.7e-01") +
  annotate("label", x =0.5, y = 2, label = " 3.97e-01 ") +
  annotate("label", x =0.5, y = 1, label = " 9.79e-61")
  
```

### UPF1 and UPF3B DEGs 


```{r}
upf3b_cor = upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,"EnsID" = ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(lists$upf3b %>% 
               dplyr::filter( padj < 0.05) %>%
                dplyr::select(EnsID, UPF3B.logFC=log2FoldChan), 
             by = "EnsID") %>% 
  ggscatter(., x="logFC", y = "UPF3B.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF3B LCLs (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 21 overlapping genes")

my_gg = upf3b_cor  + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)

```


### UPF1 and UPF2 CNVs DEGs 


```{r}
upf2_cor = upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,"EnsID" = ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(lists$upf2_group %>% 
               dplyr::filter( padj < 0.05) %>%
                dplyr::select(EnsID, UPF2.logFC=log2FoldChang), 
             by = "EnsID") %>% 
  ggscatter(., x="logFC", y = "UPF2.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF2 CNVs LCLs (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 36 overlapping genes")

my_gg = upf2_cor + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)

```

### UPF1 and UPF2 Frameshift DEGs 

```{r}
upf2_fs = upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,"EnsID" = ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(lists$upf2_frameshift %>% 
               dplyr::filter( padj < 0.05) %>%
                dplyr::select(EnsID, UPF2.logFC=log2FoldChange), 
             by = "EnsID") %>% 
  ggscatter(., x="logFC", y = "UPF2.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("UPF2 Frameshift LCL (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 5 overlapping genes")

my_gg = upf2_fs  + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```

## UPF1 DEGs and SMG8/9 LCLs

```{r}
smg_cor = upf1_results_lfc %>% 
  dplyr::filter(adj.P.Val < 0.05) %>%
  dplyr::select(SYMBOL,"EnsID" = ensembl_gene_id, logFC, adj.P.Val) %>% 
  inner_join(lists$smg %>% 
               dplyr::filter( padj < 0.05) %>%
                dplyr::select(EnsID, SMG.logFC=log2FoldChange), 
             by = "EnsID") %>% 
  ggscatter(., x="logFC", y = "SMG.logFC", cor.coef = TRUE, add = "reg.line", 
              size=5, alpha =0.6, 
              conf.int = TRUE, add.params = list(color = "#EF3829",
                                                 fill = "lightgray")) + 
    theme_bw() + ylab("SMG8/9 (log2FoldChange)") + xlab("UPF1 (log2FoldChange)") +
  geom_hline(yintercept = 0, size = 0.5,lty = "dashed", color = "grey60") +
  geom_vline(xintercept = 0, size = 0.5, lty = "dashed", color = "grey60") + ggtitle("Direction of expression of the 124 overlapping genes")

my_gg = smg_cor + geom_point_interactive(aes(tooltip =SYMBOL, data_id = SYMBOL), 
    size = 1, hover_nearest = TRUE)
girafe(ggobj = my_gg)
```



## Correlation with FRAX DEGs 


```{r}
do.call(rbind, overlap_sig_frax) %>% 
  as.data.frame() %>%
  mutate(overlap = c("UPF3B LCL DEGs", 
                     "UPF2 CNVs LCL DEGs", 
                     "UPF2 Frameshift LCLs", 
                     "SMG8/9 LCLs DEGs")) %>%
ggplot(aes(x=enrichment_nif_OR, y = overlap)) + geom_bar(stat = "identity", fill = "#edafb8") +
  theme_bw() + geom_vline(xintercept = 1) + 
   theme(axis.text.x = element_text(size = 12, family = "serif", color = "black"),
            axis.text.y = element_text(size = 12, family = "serif", color = "black"),
        axis.title = element_text(size=12, family = "serif"), 
         legend.box.background = element_rect(color = "black"),
          legend.text = element_text(family = "serif"), 
          legend.title = element_text( family = "serif"),
           plot.title = element_text(family = "serif", size =20),         
        panel.grid.major.x = element_blank(),
        panel.grid.major.y = element_line( size=.1 ),
              legend.position = "top",
        strip.text.y = element_text(
        size = 10, face = "bold.italic", family = "serif"
        ),
         strip.text.x = element_text(
        size = 10, family = "serif"
        )) + labs(x="Odds ratio", 
                  y = "") + 
  annotate("label", x =0.25, y = 4, label = " 0.99") +
   annotate("label", x =0.25, y = 3, label = " 0.65") +
  annotate("label", x =0.25, y = 2, label = " 0.99") +
  annotate("label", x =0.25, y = 1, label = "  0.44") 
  
```



# Data export

