---
title: "Exploratory-DEG"
author: "unawaz1996"
date: "2023-09-28"
output:
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  
chunk_output_type: console

---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```


```{r}
source("/home/neuro/Documents/NMD_analysis/Analysis/UPF1-FMR1/code/libraries.R")
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(gridExtra)
library(EnsDb.Hsapiens.v86)
library(ggrepel)
library(tibble)
```

```{r}
txdf = transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")])
```

## Introduction - Urwah's analysis

This includes my preprocessing using Salmon 

```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon/", "", salmon)
sample_names = gsub(".gz_transcripts", "", sample_names)
sample_names = gsub("\\_.*", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything()) %>% 
  mutate(names = gsub("\\_.*", "", names) )
md = md[order(match(md$names, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)

md %<>% dplyr::filter(Group != "FMR1") %>%
 dplyr::filter(names != "23-LDJ6767")

all_files = all_files[names(all_files) %in% md$names]
#comparisons = unique(md$Group)[-1]
#md$names = as.character(md$names)
```




```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= 1 ) >= 3)
```


## Principal component analysis 

- Controls only 

```{r}

md_controls = md %>% dplyr::filter(Group == "Control")
controls = txi_genes$abundance %>% 
  as.data.frame() %>% 
  dplyr::select(contains(md_controls$names))

pca = log2(controls[keep.genes,]  + 0.05) %>%
    t() %>% 
    prcomp(scale = FALSE)
```

- PCA using salmon abundance
```{r fig.cap="Principal component analysis"}
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
   rownames_to_column("names") %>%
    left_join(md, by= "names") %>% 
    as.tibble() %>%
    ggplot(aes(PC1, PC2, color = Group, shape = as.character(Batch))) +
    geom_point(size = 8) +
    theme_bw() + 
   geom_text_repel(aes(label = Sample.ID), show.legend = FALSE) +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))

gene_pca
```

```{r}
log_gene_cor = cor(log2(controls[keep.genes,]  + 0.05))
log_gene_cor[log_gene_cor==1] <- NA
gene = rowMeans(log_gene_cor, na.rm = TRUE) %>% scale() %>% as.data.frame() %>%
  set_colnames("gene")

gene %>% 
  rownames_to_column("names") %>%
  left_join(md, by = "names") %>%
  DT::datatable()
```



- PCA using DESeq2 VST transformation as in Clare 

```{r}
dds <- DESeqDataSetFromTximport(txi_genes,
                                colData = md,
                                design = ~ Group + Batch + Sex)
```

```{r}
smallestGroupSize <- 2
keep <- rowSums(counts(dds) >= 1) >= smallestGroupSize
dds <- dds[keep,]
vsd <- vst(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("Group", "Sample.ID"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()  +  geom_text_repel(aes(label =  Sample.ID), show.legend = FALSE) +
  theme_bw()
```


## Using DESeq2 to do a DEG analysis

```{r}
dds$Group <- factor(dds$Group, levels = c("Control","UPF1", "FRAX", "FMR1"))
```

```{r}
dds <- DESeq(dds)

res <- results(dds, contrast=c("Group","UPF1","Control"))


gene_counts = list()
thres_logFC = c( 0 ,log2(2), log2(1.5))

for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .05 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.05 ", as.character(thres))]] = results 
}



for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .005 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.005 ", as.character(thres))]] = results 
}


gene_counts %<>% 
  do.call(rbind, .) %>% 
  rownames_to_column("Threshold") %>% 
  mutate(Threshold = gsub("\\.[1-2].*", "", Threshold)) 

```


```{r}
gene_counts %>% 
  mutate(Freq = ifelse(LogFC == "Downregulated", -Freq, Freq)) %>%
  
  ggplot(aes(x= Threshold, y = Freq, fill = LogFC)) + geom_bar(stat = "identity") + 
  coord_flip() + 
   scale_x_discrete(labels = c("pvalue < 0.05 0" = "adj pvalue < 0.05, no LogFC", 
                                "pvalue < 0.05 0.584962500721156" = "adj pvalue < 0.05, abs(logfc) > log2(1.5)",
                                 "pvalue < 0.05 1" = "adj pvalue < 0.05, abs(logfc) > log2(2)",
                                "pvalue < 0.005 1" = "adj pvalue < 0.005, abs(logfc) > log2(2)",
                                 "pvalue < 0.005 0.584962500721156" = "adj pvalue < 0.005, abs(logfc) > log2(1.5)",
                                "pvalue < 0.005 0" = "adj pvalue < 0.005, no LogFC")) + theme_bw() +
  ylab("Number of DEGs") + ylim(-800, 800)
```

## Using alignment data - there may be bias due to how I aligned on using Salmon 

```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/UPF1_FMR1_for_Urwah")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/UPF1_FMR1_for_Urwah/", "", salmon)
sample_names = gsub("_transcripts", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything())
md$Sample.ID[26:27] = c("UPF1_61045", "UPF1_FD61046")
md$Sample.ID[31:32] = c("FC33_60921","FMR1_MM60955")


md = md[order(match(md$Sample.ID, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)


```

```{r}
library(VennDiagram)
x = list( "salmon" = sample_names,
          "md" = md$Sample.ID)
#md$Sample.ID[26:27] = c("UPF1_61045", "UPF1_FD61046")
#md$Sample.ID[31:32] = c("FC33_60921","FMR1_MM60955")
ggvenn(x[c(1,2)], fill_color = c("#2F124B", "#801D2E"),
       set_name_color = "black", text_color = "white", text_size = 6,
       stroke_size = 0.5, fill_alpha = 0.8,
       stroke_color = "white")


#overlap= calculate.overlap(x=list(sample_names, md$names))
```


```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
#keep.genes = (rowSums(txi_genes$abundance >= 10 ) >= 2)
```

```{r}
dds <- DESeqDataSetFromTximport(txi_genes,
                                colData = md,
                                design = ~ Sex + Batch + Group)
```

How is NMD involved in activity dependent gene expression
ADGE --> memeory and learning
What is the impact of suppressed NMD in gene expression
UPF1 was chosen to model that effect 
```{r}
#smallestGroupSize <- 2
#keep <- rowSums(counts(dds) >= 10) >= smallestGroupSize
#dds <- dds[keep,]
vsd <- vst(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("Group", "Sample.ID"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()  +  geom_text_repel(aes(label =  Sample.ID), show.legend = FALSE) +
  theme_bw()
```


```{r}
dds$Group <- factor(dds$Group, levels = c("Control","UPF1", "FRAX", "FMR1"))
```

```{r}
dds <- DESeq(dds)
res <- lfcShrink(dds, coef="Type_UPF1_vs_CTRL", type="apeglm")

res <- results(dds, contrast=c("Group","UPF1","Control"))

gene_counts = list()
thres_logFC = c( 0 ,log2(2), log2(1.5))

for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .05 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.05 ", as.character(thres))]] = results 
}



for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .005 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.005 ", as.character(thres))]] = results 
}


gene_counts %<>% 
  do.call(rbind, .) %>% 
  rownames_to_column("Threshold") %>% 
  mutate(Threshold = gsub("\\.[1-2].*", "", Threshold)) 
```

```{r}

gene_counts %>% 
  mutate(Freq = ifelse(LogFC == "Downregulated", -Freq, Freq)) %>%
  
  ggplot(aes(x= Threshold, y = Freq, fill = LogFC)) + geom_bar(stat = "identity") + 
  coord_flip() + 
   scale_x_discrete(labels = c("pvalue < 0.05 0" = "adj pvalue < 0.05, no LogFC", 
                                "pvalue < 0.05 0.584962500721156" = "adj pvalue < 0.05, abs(logfc) > log2(1.5)",
                                 "pvalue < 0.05 1" = "adj pvalue < 0.05, abs(logfc) > log2(2)",
                                "pvalue < 0.005 1" = "adj pvalue < 0.005, abs(logfc) > log2(2)",
                                 "pvalue < 0.005 0.584962500721156" = "adj pvalue < 0.005, abs(logfc) > log2(1.5)",
                                "pvalue < 0.005 0" = "adj pvalue < 0.005, no LogFC")) + theme_bw() +
  ylab("Number of DEGs") + ylim(-1000, 1000)

```
