---
title: "Quality-control"
author: "unawaz1996"
date: "2023-09-23"
output:
  html_notebook: default
  workflowr::wflow_html:
    code_folding: hide
editor_options:
  
  chunk_output_type: console
---


```{r setup, include=FALSE}
knitr::opts_chunk$set(
    autodep = TRUE,
    eval = TRUE,
	echo = TRUE,
	warning = FALSE,
	message = FALSE
)
```


```{r}
source(here::here("code/libraries.R"))
library(stargazer)
library(ggfortify)
library(glue)
library(cowplot)
library(broom)
library(glmpca)
library(naniar)
library(gridExtra)
library(EnsDb.Hsapiens.v86)
library(ggrepel)
library(org.Hs.eg.db)
```


```{r}
twoCols <- c(rgb(0.8, 0.1, 0.1), rgb(0.2, 0.2, 0.8))
```

```{r}
txdf = transcripts(EnsDb.Hsapiens.v86, return.type="DataFrame")
tx2gene = as.data.frame(txdf[,c("tx_id","gene_id", "tx_biotype")])
```


```{r}
ah <- AnnotationHub() %>%
    subset(species == "Homo sapiens") %>%
    subset(rdataclass == "EnsDb") %>% 
  subset(genome == "GRCh38")

ensDb <- ah[["AH109606"]]
grTrans <- transcripts(ensDb)
trLengths <- exonsBy(ensDb, "tx") %>%
    width() %>%
    vapply(sum, integer(1))
mcols(grTrans)$length <- trLengths[names(grTrans)]

genesGR = genes(ensDb)
transGR = transcripts(ensDb)

mcols(transGR) = mcols(transGR) %>%
  cbind(
    transcriptLengths(ensDb)[rownames(.), c("nexon", "tx_len")]
  )
```



```{r}
mcols(genesGR) <- mcols(genesGR) %>%
  as.data.frame() %>%
  dplyr::select(
    gene_id, gene_name, gene_biotype, entrezid
  ) %>%
  left_join(
    mcols(transGR) %>%
      as.data.frame() %>%
      mutate(
        tx_support_level = case_when(
          is.na(tx_support_level) ~ 1L, 
          TRUE ~ tx_support_level
        )
      ) %>%
      group_by(gene_id) %>%
      summarise(
        n_tx = n(),
        longest_tx = max(tx_len),
        ave_tx_len = mean(tx_len),
        gc_content = sum(tx_len*gc_content) / sum(tx_len)
      ) %>%
      mutate(
        bin_length = cut(
          x = ave_tx_len,
          labels = seq_len(10),
          breaks = quantile(ave_tx_len, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin_gc = cut(
          x = gc_content,
          labels = seq_len(10),
          breaks = quantile(gc_content, probs = seq(0, 1, length.out = 11)),
          include.lowest = TRUE
        ),
        bin = paste(bin_gc, bin_length, sep = "_")
      ),
    by = "gene_id"
  ) %>%
  set_rownames(.$gene_id) %>%
  as("DataFrame")

trans2Gene <- mcols(transGR) %>%
    as.data.frame() %>%
    dplyr::select(tx_id, gene_id) %>%
    dplyr::filter(!is.na(tx_id), !is.na(gene_id)) %>%
    as_tibble()

```

# FastQC summary 

```{r load-fastqc}
rawFqc <- list.files(
    path = here::here("data/fastqc/Raw"),
    pattern = "zip",
    full.names = TRUE
    ) %>%
    FastqcDataList()
# 
# trmdFqc <- list.files(
#     path = here::here("data/fastqc/Trimmed"),
#     pattern = "zip",
#     full.names = TRUE
#     ) %>%
#     FastqcDataList()


```


```{r fastq-summary, fig.height=10, fig.width=6, fig.cap="*Basic statistics summary plot. Figure (a) hsows the summary of the PASS/FAIl flags prior to base quality and adapter trimming. Figure (b) shows the summary of PASS/FAIL flags after quality trimming with trimgalore. Green: PASS; Yellow: FAIL; Red: WARN *"}
sum_raw = plotSummary(rawFqc)
#sum_trmd = plotSummary(trmdFqc)

sum_raw
```

```{r fig.width = 10, fig.cap="*Sequence Length distribution of the RNA seq reads (a) before and (b) after quality trimming. Only reads with the base length of >  150bp were retained after quality trimming.*"}
ngsReports::plotSeqLengthDistn(rawFqc[[1]])
#dist_trmd = ngsReports::plotSeqLengthDistn(trmdFqc[[1]])
#grid.arrange(dist_raw, dist_trmd, ncol=2)
```

```{r}
reads <- readTotals(rawFqc)
#reads_trimmed = readTotals(trmdFqc)
#reads %>%
  #as.data.frame() %>% 
  #dplyr::rename("Raw" = "Total_Sequences") %>% 
  #cbind(dplyr::rename(as.data.frame(reads_trimmed),"Trimmed" = "Total_Sequences", "New" = "Filename")) %>% 
  #dplyr::select(-New) %>% 
  #mutate(names = gsub("\\_.*", "", Filename)) %>% 
  #left_join(md) %>% 
  #dplyr::select(names, Group, Condition_UPF3B, Condition_UPF3A, "Label"= Please_Label, everything()) %>% 
  #stargazer(title="Sample information including number of raw reads obtained for each sample through Illumina paired-end sequencing and the #number of reads retained after adapter and quality trimming",  summary=FALSE, rownames = FALSE, out = #"output/QC/Thesis_tables/data_summary.tex")
```
## Library sizes

Library sizes for unprocessed data ranged between 86,178,215 and 129,449,694 reads.

```{r fig.height = 7, fig.width =7, fig.cap="*Total numner 0f reads from each sample (a) before and (b) after quality trimming with trimgalore.*"}
r1 <- grepl("_1", fqName(rawFqc))
r2 = grepl("_2", fqName(rawFqc))
#trmd_r1 <- grepl("_1", fqName(trmdFqc))
raw_reads = plotReadTotals(rawFqc[r1], barCols = twoCols)
#trmd_reads = plotReadTotals(trmdFqc[trmd_r1], barCols = twoCols)
#rid.arrange(raw_reads, trmd_reads, nrow=2)
#

raw_reads
```

## GC content

In poly(A) selected RNA-seq library preparation methods, the nonuniform coverage of transcripts is a prevalent issue. As poly(A) tail only occurs at the 3' end of the mRNA, this can usually result in an over-representation of the 3' end. Bias at the 5′ end of RNA can also happen because of various factors, such as the fragmentation method (the 5′ end of RNA is more stable), reverse transcription from RNA to cDNA and strand-oriented library construction protocol.

GC content allows the exploration of the sequencing coverage and can indicate issues in overrepresentation. It has been observed that either high or low GC content will result in lower depth coverage.

```{r}
gcPlots <- list(
   r1 = plotGcContent(
       x = rawFqc[r1],
       plotType = "line",
       gcType = "Transcriptome",
       species = "Hsapiens"
   ),
   r2 = plotGcContent(
       x = rawFqc[!r1],
       plotType = "line",
       gcType = "Transcriptome",
       species = "Hsapiens"
   )
)
lg <- get_legend(gcPlots$r2 + theme(legend.position = "bottom"))
# pdf(file ="output/QC/Thesis_figures/Transcript_GC.pdf", width = 7.52, height = 5.83)
plot_grid(
   plot_grid(
       r1 = gcPlots$r1 +
           ggtitle("R1: GC Distribution", subtitle = c()) +
           theme(legend.position = "none"),
       r2 = gcPlots$r2 +
           ggtitle("R2: GC Distribution", subtitle = c()) +
           theme(legend.position = "none")
   ),
   lg = lg,
   nrow = 2,
   rel_heights = c(5,2)
)
# dev.off

```

### Per base sequence quality
```{r}

pbsQ_all = plotBaseQuals(rawFqc)
pbsQ_sample = plotSeqQuals(rawFqc[[1]])


pbsQ_sample
```

```{r}
pbsQ_all
```

### Per base sequence content 

```{r}
plotSeqContent(rawFqc[[1]])
plotSeqContent(rawFqc)
plotSeqContent(rawFqc[1:2], plotType = "line", nc = 1)
```



<!-- ## Trimmed data -->

<!-- ```{r} -->
<!-- trimStats = readTotals(rawFqc) %>% -->
<!--     dplyr::rename(Raw = Total_Sequences, Name = Filename) %>% -->
<!--     cbind(readTotals(trmdFqc)) %>% -->
<!--     dplyr::rename(Trimmed = Total_Sequences) %>%  -->
<!--   dplyr::select(-Filename) %>% -->
<!--     dplyr::filter(grepl("_1", Name)) %>% -->
<!--     mutate( -->
<!--         Discarded = 1 - Trimmed/Raw, -->
<!--         Retained = Trimmed / Raw -->
<!--     ) -->
<!-- ``` -->

<!-- ```{r} -->
<!-- reads %>% -->
<!--   as.data.frame() %>%  -->
<!--   dplyr::rename("Raw" = "Total_Sequences") %>%  -->
<!--   cbind(dplyr::rename(as.data.frame(reads_trimmed),"Trimmed" = "Total_Sequences", "New" = "Filename")) %>%  -->
<!--   cbind(dplyr::select(as.data.frame(trimStats), Discarded, Retained)) %>%  -->
<!--   dplyr::select(-New) %>%  -->
<!--   mutate(names = gsub("\\_.*", "", Filename)) %>%  -->
<!--   left_join(md) %>%  -->
<!--   dplyr::select(names, Group, Condition_UPF3B, Condition_UPF3A, "Label"= Please_Label, everything(), -files) %>%  -->
<!--   pander(split.tables = Inf) -->
<!-- ``` -->

<!-- After adapter trimming, \< 1% of reads were discarded. -->


## Alignment 
Over here is where I perform initial QC analyses including a principal component analysis and an initial exploratory DEG analysis

```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon/", "", salmon)
sample_names = gsub(".gz_transcripts", "", sample_names)
sample_names = gsub("\\_.*", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything()) %>% 
  mutate(names = gsub("\\_.*", "", names) )
md = md[order(match(md$names, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)

md %<>% dplyr::filter(Group != "FMR1") %>%
 dplyr::filter(names != "23-LDJ6767")

all_files = all_files[names(all_files) %in% md$names]
```

```{r transcrpt}
se <- tximeta(md)
keep = (rowSums(se@assays@data$abundance >= 5) >= 3)

```

```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= 1 ) >= 3)
```


```{r}
group_cols <- hcl.colors(
  n = length(unique(md$Group)), 
  palette = "Zissou 1"
  ) %>%
  setNames(unique(md$Group))
```

## Genotype check 

```{r, fig.height = 6, fig.width=10, fig.cap="All samples demonstrated the expected expression patterns, no mislabeling was detected in the dataset."}
group.labs <- c("Control", "FRAX", "UPF1")
names(group.labs) <- unique(md$Group)

txi_genes$abundance["ENSG00000102081",] %>% melt() %>%
   rownames_to_column("names") %>% left_join(md, by = "names") %>%
   ggplot(aes(x=as.character(names), y= value)) +
   geom_bar(stat="identity", color = "black", alpha = 0.65,  fill = "#E82F61") +
   facet_grid(~Group, scales = "free_x") + ylab("tpm") + theme_bw() + xlab("") +
#   scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD",
 #                                "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE",
  #                               "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
   theme(legend.position = "none", axis.text.x=element_blank()) + ggtitle("FMR1 ENSG00000102081")
```


```{r, fig.height = 6, fig.width=10, fig.cap="All samples demonstrated the expected expression patterns, no mislabeling was detected in the dataset."}

txi_genes$abundance["ENSG00000005007",] %>% melt() %>%
   rownames_to_column("names") %>% left_join(md, by = "names") %>%
   ggplot(aes(x=as.character(names), y= value)) +
   geom_bar(stat="identity", color = "black", alpha = 0.65,  fill = "#E82F61") +
   facet_grid(~Group, scales = "free_x") + ylab("tpm") + theme_bw() + xlab("") +
#   scale_fill_manual(labels = c("UPF3A_KD_UPF3B_KD" = "Double KD", "UPF3A_KD" = "UPF3A KD",
 #                                "UPF3B_KD" = "UPF3B KD", "UPF3A_OE" = "UPF3A OE",
  #                               "UPF3A_OE_UPF3B_KD" = "UPF3A OE UPF3B KD")) +
   theme(legend.position = "none", axis.text.x=element_blank()) + ggtitle("UPF1 ENSG00000005007")



```


```{r fig.height = 4, fig.cap="Expression density plots for all samples before and after filtering, showing logCPM values."}
exp_den = txi_genes$counts %>% cpm(log=TRUE) %>%
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("Before filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + 
    theme_bw() +
    theme(legend.position = "none")

exp_den_filt = txi_genes$counts %>%
    cpm(log=TRUE) %>%
    magrittr::extract(keep.genes,) %>% 
    melt() %>%
    dplyr::filter(is.finite(value)) %>% 
    ggplot(aes(x=value, color = as.character(Var2))) +
    geom_density() +
    ggtitle("After filtering") +
    labs(x = "logCPM", y = "Proportion of Genes") + theme_bw() +
    theme(legend.position = "none")

grid.arrange(exp_den,exp_den_filt, ncol=2 )
```


```{r transcript-level}
dge = se@assays@data$counts[keep,] %>% 
  as.data.frame %>% 
  rownames_to_column("tx_id") %>% 
  set_colnames(basename(colnames(.))) %>% 
  dplyr::filter(tx_id %in% trans2Gene$tx_id) %>% 
  pivot_longer(cols = all_of(md$names), names_to = "Hs", values_to = "count") %>% 
    left_join(trans2Gene) %>%
    group_by(Hs, gene_id) %>%
    summarise(count = sum(count)) %>% 
  pivot_wider(names_from = "Hs", values_from = "count") %>% 
    dplyr::filter(grepl("ENSG00", gene_id)) %>%
    as.data.frame() %>% 
    column_to_rownames("gene_id") %>% 
    DGEList() 
dge$samples %<>%
    mutate(names = rownames(.)) %>%
  dplyr::select(-group) %>% 
    left_join(md, by = "names") %>%
    set_rownames(.$names)
dge$genes <- genesGR[rownames(dge)] %>% 
  mcols()
```



## Library size

```{r}
dge$samples %>% 
  mutate(lib.size = lib.size / 1e6) %>% 
  ggplot(aes(Sample.ID , lib.size, fill = Group)) +
  geom_col() +
  facet_wrap(~Group, scales = "free_x") +
  scale_y_continuous(expand = expansion(c(0, .05))) +
  scale_fill_manual(values = group_cols) +
  labs(
    x = "Sample", y = "Library Size (millions)",
    fill = "Group"
  ) + theme_bw() + theme(legend.position = "none",
                         axis.text.x = element_text(angle = 90)) 
```

## Counts assingment rate

<!-- ```{r} -->
<!-- trmdFqc %>%  -->
<!--   getModule("Basic") %>%  -->
<!--   mutate(names = gsub("\\_.*", "", Filename)) %>%  -->
<!--   left_join(dge$samples, by = "names") %>%  -->
<!--   mutate(`% Assigned To Genes` = lib.size / Total_Sequences) %>%  -->
<!--   ggplot(aes(names, `% Assigned To Genes`, fill = Group)) + -->
<!--   geom_col() + -->
<!--   facet_wrap(~Group, scales = "free_x") + -->
<!--   scale_fill_manual(values = group_cols) + -->
<!--   scale_y_continuous(labels = percent, expand = expansion(c(0, 0.05))) + -->
<!--   labs(fill = "Group") -->
<!-- ``` -->

## Total detected genes

```{r}
dge$counts %>%
  as_tibble() %>%
  mutate(
    across(everything(), as.logical)
  ) %>%
  summarise(
    across(everything(), sum)
  ) %>%
  pivot_longer(
    everything(), names_to = "names", values_to = "Detected"
  ) %>%
  left_join(md)%>%
  ggplot(aes(Group, Detected, colour = Group)) +
  geom_point() +
  geom_segment(
    aes(xend = Group, y = 0, yend = Detected),
    data = . %>% 
      group_by(Group) %>%
      summarise(Detected = min(Detected)),
    colour = "black", size = 1/4) +
  scale_y_continuous(labels = comma, expand = expansion(c(0, 0.05))) +
  scale_colour_manual(values = group_cols) +
  labs(
    x = "Group", 
    y = "Genes Detected",
    colour = "Group"
  )

```

## Transcript level exploration of data

```{r}
## Filtering threshold
before =  se@assays@data$abundance %>% as.data.frame() %>%
    mutate(rowMeans = rowMeans(.)) %>% 
  rownames_to_column("tx_id") %>%
  left_join(tx2gene, by = "tx_id") %>%
  group_by(gene_id) %>% mutate(ntx = n()) %>% 
dplyr::select(gene_id, tx_id, tx_biotype, rowMeans, ntx) 

after = se@assays@data$abundance[keep,] %>% as.data.frame() %>%
    mutate(rowMeans = rowMeans(.)) %>% 
  rownames_to_column("tx_id") %>%
  left_join(tx2gene, by = "tx_id") %>%
  group_by(gene_id) %>% mutate(ntx = n()) %>% 
dplyr::select(gene_id, tx_id, tx_biotype, rowMeans, ntx) 


df <- gdata::combine(before, after) %>% dplyr::rename(Filter=source)

dplyr::group_by(df,Filter) %>% dplyr::summarise(median(rowMeans),sd(rowMeans))


ntxInfo  <- df %>% dplyr::group_by(Filter) %>%
    dplyr::distinct(gene_id,.keep_all = TRUE)  
biotypeCount <- df %>% dplyr::group_by(Filter) %>%
    dplyr::count(tx_biotype) %>%
    mutate(sum=sum(n),frac = n / sum(n)) 

```

```{r fig.height=6, fig.width=6, fig.cap= "*Log10 of the mean TPMs (transcript per million) over all samples before and after filtering out low-expressed transcripts and genes*"}
distribution = ggplot(df, aes(x = log10(rowMeans), col = Filter)) +
    geom_histogram(aes(y = ..density.., fill = Filter), alpha = 0.4, bins = 100) +
    geom_density(size = 1.1) +
    scale_x_continuous(name = "log10(mean TPM)", limits = c(-8,10)) +
    theme_bw() +
    scale_color_manual(values = c("#FCC98A", "#DF1861"))+
    scale_fill_manual(values = c("#FCC98A", "#DF1861"))
distribution
#ggsave(plot = dist, file = "/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-distribution.svg", 
 #      height = 4.29, width = 8.02, units = "in")
```

```{r fig.height=6, fig.width=6, fig.cap="*Violin plots showing the distribution of the number of transcripts per gene (in logarithmic scale). Violin width is scaled by the total number of observations while jittered points represent actual observations.*"}
biocount = ggplot(subset(biotypeCount, frac >= 0.001), aes(x = as.factor(tx_biotype), y = frac, fill = Filter)) +
    geom_bar( stat = "identity") +
    coord_flip() +
    facet_wrap(.~Filter) +
    theme_bw() +
    theme(axis.text.x = element_text(angle = -90)) +
    labs(x = "Transcript biotype (Ensembl v75)", y = "Frequency") +
        scale_color_manual(values = c("#FCC98A", "#DF1861"))+
    scale_fill_manual(values = c("#FCC98A", "#DF1861"))+  
theme(panel.spacing = unit(1, "lines")) +
theme(strip.background = element_rect(fill = "black"))+
theme(strip.text = element_text(color = "white", face = "bold"))
biocount

#ggsave(plot = biocount, file ="/home/neuro/Documents/NMD_analysis/Analysis/NMD-analysis/output/Transcript/Thesis_figures/transcript-composition.svg", 
 #      height = 4.05, width = 7.74, units = "in")
```

```{r  fig.height=6, fig.width=6, fig.cap="*Violin plots showing the distribution of the number of transcripts per gene (in logarithmic scale). Violin width is scaled by the total number of observations while jittered points represent actual observations.*"}
gene_info =ggplot(ntxInfo, aes(x = Filter, y = log10(ntx))) +
    geom_violin(scale = "count", aes(fill = Filter)) +
    geom_jitter(width = 0.1, size = 0.3, alpha = 0.01) +
    theme_bw() +
    scale_y_continuous(name = "log10(# transcripts per gene)") + 
    labs(x = "") +
        scale_color_manual(values = c("deepskyblue3", "darkblue"))+
    scale_fill_manual(values = c("deepskyblue3", "darkblue"))   
gene_info
```



<!-- # ```{r} -->
<!-- # library(VennDiagram) -->
<!-- # x = list( "salmon" = sample_names,  -->
<!-- #           "md" = md$GeneWiz.ID) -->
<!-- #  -->
<!-- # venn_upf3 = ggvenn(x[c(1,2)], fill_color = c("#2F124B", "#801D2E"), -->
<!-- #        set_name_color = "black", text_color = "white", text_size = 6,  -->
<!-- #        stroke_size = 0.5, fill_alpha = 0.8,  -->
<!-- #        stroke_color = "white")  -->
<!-- #  -->
<!-- # #md$names[26:27] = c("UPF1_61045", "UPF1_FD61046") -->
<!-- # #md$names[31:32] = c("FC33_60921","FMR1_MM60955") -->
<!-- # #overlap= calculate.overlap(x=list(sample_names, md$names)) -->
<!-- # ``` -->

## Principal component analysis 

```{r}
pca = log(assays(se)$abundance[keep,] + 0.05) %>%
  t() %>% 
  prcomp(scale = FALSE)
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

#showLabel = nrow(md) <= 18
trans_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
  rownames_to_column("names") %>%
    left_join(md, by = "names") %>% 
    as.tibble() %>%
    ggplot(aes(PC1, PC2, color = Group, shape = as.character(Batch)), size = lib.size/1e6) +
    geom_point(size = 8) +
    theme_bw() + 
   geom_text_repel(aes(label = Sample.ID), show.legend = FALSE) +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))
  
```


```{r}
pca = log2(txi_genes$abundance[keep.genes,]  + 0.05) %>%
    t() %>% 
    prcomp(scale = FALSE)
```

- PCA using salmon abundance
```{r fig.cap="Principal component analysis"}
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
  rownames_to_column("names") %>%
    left_join(md, by = "names") %>% 
    as.tibble() %>%
    ggplot(aes(PC1, PC2, color = Group, shape = as.character(Batch))) +
    geom_point(size = 8) +
    theme_bw() + 
   geom_text_repel(aes(label = Sample.ID), show.legend = FALSE) +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))

gene_pca
```


```{r fig.height = 10, fig.width=15}
pca_all = ggarrange(gene_pca, trans_pca , ncol=2,  common.legend = TRUE, legend="right")
```






```{rfig.height = 10, fig.width=1, fig.cap="Initial pricipal component analysis of gene (left) and transcript (right) level data"}
pca_all
```

- PCA using DESeq2 VST transformation as in Clare 

```{r}
dds <- DESeqDataSetFromTximport(txi_genes,
                                colData = md,
                                design = ~ Sex + Batch + Group)
```

```{r fig.height=6, fig.width=6, fig.cap="PCA analysis on gene level after after performing VST from DESeq2 package"}
smallestGroupSize <- 2
keep <- rowSums(counts(dds) >= 1) >= smallestGroupSize
dds <- dds[keep,]
vsd <- vst(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("Group", "Sample.ID"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()  +  geom_text_repel(aes(label =  Sample.ID), show.legend = FALSE) +
  theme_bw()
```

<!-- # ```{r} -->
<!-- # gene_pca = pca$x[,2:3] %>%  -->
<!-- #     as.data.frame()  %>% -->
<!-- #     cbind(md) %>% -->
<!-- #     as.tibble() %>% -->
<!-- #     ggplot(aes(PC2, PC3, color = Group)) + -->
<!-- #     geom_point(size = 11.5) + -->
<!-- #   geom_text_repel(aes(label = names), show.legend = FALSE) + -->
<!-- #     theme_bw() +  -->
<!-- #     theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12),  -->
<!-- #           axis.text = element_text(size = 10)) + -->
<!-- #   labs(x = paste0("PC2 (", pcaVars[["PC2"]], ")"), -->
<!-- #     y = paste0("PC3 (", pcaVars[["PC3"]], ")")) -->
<!-- # gene_pca  -->
<!-- # ``` -->

- Correlation of samples 

```{r}
log_gene_cor = cor(log2(txi_genes$abundance[keep.genes,]  + 0.05))
log_gene_cor[log_gene_cor==1] <- NA
gene = rowMeans(log_gene_cor, na.rm = TRUE) %>% scale() %>% as.data.frame() %>%
  set_colnames("gene")

gene %>% 
  rownames_to_column("names") %>%
  left_join(md, by = "names") %>%
  DT::datatable(caption = "Correlation of the mean of the standard deviation of all samples")
```



- Controls only 

```{r}

md_controls = md %>% dplyr::filter(Group == "Control")
controls = txi_genes$abundance %>% 
  as.data.frame() %>% 
  dplyr::select(contains(md_controls$names))

pca = log2(controls[keep.genes,]  + 0.05) %>%
    t() %>% 
    prcomp(scale = FALSE)
```


```{r fig.cap="Gene level principal component analysis of controls"}
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
   rownames_to_column("names") %>%
    left_join(md, by= "names") %>% 
    as.tibble() %>%
    ggplot(aes(PC1, PC2, color = Group, shape = as.character(Batch))) +
    geom_point(size = 8) +
    theme_bw() + 
   geom_text_repel(aes(label = Sample.ID), show.legend = FALSE) +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))

gene_pca
```


```{r}
log_gene_cor = cor(log2(controls[keep.genes,]  + 0.05))
log_gene_cor[log_gene_cor==1] <- NA
gene = rowMeans(log_gene_cor, na.rm = TRUE) %>% scale() %>% as.data.frame() %>%
  set_colnames("gene")

gene %>% 
  rownames_to_column("names") %>%
  left_join(md, by = "names") %>%
  DT::datatable(caption = "Correlation of the mean of the standard deviation of controls")
```

- Removing the outlier sample from the analysis 
```{r}
salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon")
salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE)
all_files = file.path(salmon, "quant.sf")
sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/Salmon/", "", salmon)
sample_names = gsub(".gz_transcripts", "", sample_names)
sample_names = gsub("\\_.*", "", sample_names)
names(all_files) <- sample_names
md = read.csv(here::here("data/Sample_info.csv"), header= TRUE) %>%  
  #mutate(files = file.path(salmon, "quant.sf")) %>% 
  dplyr::rename("names" = "GeneWiz.ID", 
                "Group" = "Sample.type") %>% 
  mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>%
  dplyr::select(names,everything()) %>% 
  mutate(names = gsub("\\_.*", "", names) )
md = md[order(match(md$names, sample_names)),]
md %<>% 
  rownames_to_column("random") %>% 
#  column_to_rownames("names") %>%
  dplyr::select(-random) %>%
  mutate(files = all_files)

md %<>% dplyr::filter(Group != "FMR1") %>%
 dplyr::filter(names != "23-LDJ6767") %>%
dplyr::filter(names != "202")
all_files = all_files[names(all_files) %in% md$names]
```


```{r}
txi_genes =  tximport(all_files, type="salmon", txOut=FALSE,
                      countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE)
keep.genes = (rowSums(txi_genes$abundance >= 1 ) >= 3)
```

```{r}
pca = log2(txi_genes$abundance[keep.genes,]  + 0.05) %>%
    t() %>% 
    prcomp(scale = FALSE)
```


```{r fig.cap="Principal component analysis after removing outlier control sample"}
pcaVars <- percent_format(0.1)(summary(pca)$importance["Proportion of Variance",]) 

gene_pca = pca$x[,1:2] %>% 
    as.data.frame()  %>%
  rownames_to_column("names") %>%
    left_join(md, by = "names") %>% 
    as.tibble() %>%
    ggplot(aes(PC1, PC2, color = Group, shape = as.character(Batch))) +
    geom_point(size = 8) +
    theme_bw() + 
   geom_text_repel(aes(label = Sample.ID), show.legend = FALSE) +
    theme(axis.title.y = element_text(size = 12), axis.title.x = element_text(size = 12), 
          axis.text = element_text(size = 10)) +
  labs(x = paste0("PC1 (", pcaVars[["PC1"]], ")"),
    y = paste0("PC2 (", pcaVars[["PC2"]], ")"))

gene_pca
```




## Exploratory DEG analysis 

- Using voom and limma 

```{r}
txi_genes_filtered = txi_genes$counts[keep.genes,]
y <- DGEList(txi_genes_filtered)


design <- model.matrix(~Batch +Group + Sex, data = md) %>% 
    set_colnames(gsub(pattern = "Group", replacement ="", x = colnames(.)))

y <- calcNormFactors(y)
v <- voom(y, design)

fit = lmFit(v, design) %>% 
    eBayes()


summary(decideTests(fit, lfc =1))
```


```{r}
upf1_results_lfc = topTable(fit,coef = "UPF1", number = Inf) %>% 
                     mutate(res = ifelse(logFC > log2(2) & adj.P.Val < 0.05, "Upregulated", 
                      ifelse(logFC < -log2(2) & adj.P.Val < 0.05, "Downregulated", "NotSig"))) %>% 
  rownames_to_column("ensembl_gene_id") %>%
   mutate(SYMBOL = mapIds(org.Hs.eg.db, keys=ensembl_gene_id,  column="SYMBOL",keytype="ENSEMBL", multiVals="first"))
```

```{r}
upf1_results_lfc %>% DT::datatable()
```






<!-- ## Using alignment data - there may be bias due to how I aligned on using Salmon  -->

<!-- ```{r} -->
<!-- salmon.files = ("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/UPF1_FMR1_for_Urwah") -->
<!-- salmon = list.files(salmon.files, pattern = "transcripts$", full.names = TRUE) -->
<!-- all_files = file.path(salmon, "quant.sf") -->
<!-- sample_names = gsub("/home/neuro/Documents/NMD_analysis/Analysis/Results/UPF1-FMR1/UPF1_FMR1_for_Urwah/", "", salmon) -->
<!-- sample_names = gsub("_transcripts", "", sample_names) -->
<!-- names(all_files) <- sample_names -->
<!-- md = read.csv("data/Sample_info.csv", header= TRUE) %>%   -->
<!--   #mutate(files = file.path(salmon, "quant.sf")) %>%  -->
<!--   dplyr::rename("names" = "GeneWiz.ID",  -->
<!--                 "Group" = "Sample.type") %>%  -->
<!--   mutate(Group = ifelse(Group == "MC" | Group == "FC", "Control", Group)) %>% -->
<!--   dplyr::select(names,everything()) -->
<!-- md$Sample.ID[26:27] = c("UPF1_61045", "UPF1_FD61046") -->
<!-- md$Sample.ID[31:32] = c("FC33_60921","FMR1_MM60955") -->


<!-- md = md[order(match(md$Sample.ID, sample_names)),] -->
<!-- md %<>%  -->
<!--   rownames_to_column("random") %>%  -->
<!-- #  column_to_rownames("names") %>% -->
<!--   dplyr::select(-random) %>% -->
<!--   mutate(files = all_files) -->


<!-- ``` -->

<!-- # ```{r} -->
<!-- # library(VennDiagram) -->
<!-- # x = list( "salmon" = sample_names, -->
<!-- #           "md" = md$Sample.ID) -->
<!-- # #md$Sample.ID[26:27] = c("UPF1_61045", "UPF1_FD61046") -->
<!-- # #md$Sample.ID[31:32] = c("FC33_60921","FMR1_MM60955") -->
<!-- # ggvenn(x[c(1,2)], fill_color = c("#2F124B", "#801D2E"), -->
<!-- #        set_name_color = "black", text_color = "white", text_size = 6, -->
<!-- #        stroke_size = 0.5, fill_alpha = 0.8, -->
<!-- #        stroke_color = "white") -->
<!-- #  -->
<!-- #  -->
<!-- # #overlap= calculate.overlap(x=list(sample_names, md$names)) -->
<!-- # ``` -->


<!-- # ```{r} -->
<!-- # txi_genes =  tximport(all_files, type="salmon", txOut=FALSE, -->
<!-- #                       countsFromAbundance="scaledTPM", tx2gene = tx2gene, ignoreTxVersion = TRUE, ignoreAfterBar = TRUE) -->
<!-- # keep.genes = (rowSums(txi_genes$abundance >= 10 ) >= 2) -->
<!-- # ``` -->
 - Using DESeq2 
 
```{r}
dds <- DESeqDataSetFromTximport(txi_genes,
                                colData = md,
                                design = ~Batch + Group + Sex)
```

```{r}
smallestGroupSize <- 2
keep <- rowSums(counts(dds) >= 10) >= smallestGroupSize
dds <- dds[keep,]
vsd <- vst(dds, blind=FALSE)
pcaData <- plotPCA(vsd, intgroup=c("Group", "Sample.ID"), returnData=TRUE)
percentVar <- round(100 * attr(pcaData, "percentVar"))
ggplot(pcaData, aes(PC1, PC2, color=Group)) +
  geom_point(size=3) +
  xlab(paste0("PC1: ",percentVar[1],"% variance")) +
  ylab(paste0("PC2: ",percentVar[2],"% variance")) + 
  coord_fixed()  +  geom_text_repel(aes(label =  Sample.ID), show.legend = FALSE) +
  theme_bw()
```


```{r}
dds$Group <- factor(dds$Group, levels = c("Control","UPF1", "FRAX", "FMR1"))
```



```{r}
dds <- DESeqDataSetFromTximport(txi_genes,
                                colData = md,
                                design = ~Batch +Group + Sex)
dds <- DESeq(dds)

res <- results(dds, contrast=c("Group","UPF1","Control"))


gene_counts = list()
thres_logFC = c( 0 ,log2(2), log2(1.5))

for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .05 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.05 ", as.character(thres))]] = results 
}



for (thres in thres_logFC){
  results = res  %>% as.data.frame %>% 
   dplyr::filter(padj < .005 & abs(log2FoldChange) > thres) %>%  
  with(.,  
     table(sign.lfc=sign(log2FoldChange))) %>% 
    as.data.frame() %>% 
    mutate(LogFC = ifelse(sign.lfc == 1, "Upregulated", "Downregulated")) %>% 
    dplyr::select(-sign.lfc)
  
  gene_counts[[paste0("pvalue < 0.005 ", as.character(thres))]] = results 
}


gene_counts %<>% 
  do.call(rbind, .) %>% 
  rownames_to_column("Threshold") %>% 
  mutate(Threshold = gsub("\\.[1-2].*", "", Threshold)) 

```

```{r fig.height=6, fig.cap="Number of up and down DEGs with different thresholds"}
gene_counts %>% 
  mutate(Freq = ifelse(LogFC == "Downregulated", -Freq, Freq)) %>%
  
  ggplot(aes(x= Threshold, y = Freq, fill = LogFC)) + geom_bar(stat = "identity") + 
  coord_flip() + 
   scale_x_discrete(labels = c("pvalue < 0.05 0" = "adj pvalue < 0.05, no LogFC", 
                                "pvalue < 0.05 0.584962500721156" = "adj pvalue < 0.05, abs(logfc) > log2(1.5)",
                                 "pvalue < 0.05 1" = "adj pvalue < 0.05, abs(logfc) > log2(2)",
                                "pvalue < 0.005 1" = "adj pvalue < 0.005, abs(logfc) > log2(2)",
                                 "pvalue < 0.005 0.584962500721156" = "adj pvalue < 0.005, abs(logfc) > log2(1.5)",
                                "pvalue < 0.005 0" = "adj pvalue < 0.005, no LogFC")) + theme_bw() +
  ylab("Number of DEGs") 
```

<!-- ```{r} -->
<!-- dds <- DESeq(dds) -->

<!-- res <- results(dds, contrast=c("Group","UPF1","Control")) -->



<!-- res  %>% as.data.frame %>% -->
<!--    dplyr::filter(padj < .05) %>% -->
<!--   with(., -->
<!--      table(sign.lfc=sign(log2FoldChange))) %>% as.data.frame() -->

<!-- res  %>% as.data.frame %>% -->
<!--    dplyr::filter(padj < .05 & abs(log2FoldChange) > log2(1.5)) %>% -->
<!--   with(., -->
<!--      table(sign.lfc=sign(log2FoldChange))) %>% as.data.frame() -->

<!-- res  %>% as.data.frame %>% -->
<!--    dplyr::filter(padj < .05 & abs(log2FoldChange) > log2(2)) %>% -->
<!--   with(., -->
<!--      table(sign.lfc=sign(log2FoldChange))) %>% as.data.frame() -->

<!-- res  %>% as.data.frame %>% -->
<!--    dplyr::filter(padj < .005 ) %>% -->
<!--   with(., -->
<!--      table(sign.lfc=sign(log2FoldChange))) %>% as.data.frame() -->

<!-- res  %>% as.data.frame %>% -->
<!--    dplyr::filter(padj < .005 & abs(log2FoldChange) > log2(2)) %>% -->
<!--   with(., -->
<!--      table(sign.lfc=sign(log2FoldChange))) %>% as.data.frame() -->
<!-- ``` -->
